<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>HaDeX: Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass Spectrometry Data • HaDeX</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="HaDeX: Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass Spectrometry Data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HaDeX</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/HaDeX.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/data_input.html">Data input</a></li>
    <li><a class="dropdown-item" href="../articles/datafiles.html">Data processing</a></li>
    <li><a class="dropdown-item" href="../articles/glossary.html">Glossary</a></li>
    <li><a class="dropdown-item" href="../articles/mass_comparison.html">Mass calculation</a></li>
    <li><a class="dropdown-item" href="../articles/quality_control.html">Quality control</a></li>
    <li><a class="dropdown-item" href="../articles/transformation.html">Data aggregation</a></li>
    <li><a class="dropdown-item" href="../articles/usage.html">Code usage</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Data visualization</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="HaDeX_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="HaDeX_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="HaDeX_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="HaDeX_files/datatables-binding-0.33/datatables.js"></script><link href="HaDeX_files/dt-core-bootstrap-1.13.6/css/dataTables.bootstrap.min.css" rel="stylesheet">
<link href="HaDeX_files/dt-core-bootstrap-1.13.6/css/dataTables.bootstrap.extra.css" rel="stylesheet">
<script src="HaDeX_files/dt-core-bootstrap-1.13.6/js/jquery.dataTables.min.js"></script><script src="HaDeX_files/dt-core-bootstrap-1.13.6/js/dataTables.bootstrap.min.js"></script><link href="HaDeX_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="HaDeX_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>HaDeX: Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass Spectrometry Data</h1>
                        <h4 data-toc-skip class="author">Weronika
Puchala, Michal Burdukiewicz, Michal Kistowski, Michal Dadlez</h4>
            
      

      <div class="d-none name"><code>HaDeX.Rmd</code></div>
    </div>

    
    
<p>HaDeX is an R package for analysis of HDX-MS data. Although all
functionalities are available in the form of a web server and a
standalone tool, some of the vignettes are more directed towards the
command-line usage of HaDeX.</p>
<div class="section level2">
<h2 id="general-resources">General resources<a class="anchor" aria-label="anchor" href="#general-resources"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<a href="hdxms.Rmd">Basics of HDX-MS</a>: Fundamental principles of
the analysis of HDX-MS data.</li>
<li>
<a href="glossary.html">Glossary</a>: all definitions and equations
necessary to understand the theoretical basics of HaDeX.</li>
<li>
<a href="visualisation.html">Data visualization</a>: visualisations
of HDX-MS data implemented in HaDeX.</li>
<li>
<a href="datafiles.html">Data usage</a>: an in-depth description how
HaDeX interprets input data from DynamX 3.0 and 2.0.</li>
</ol>
</div>
<div class="section level2">
<h2 id="resources-for-the-command-line-usage">Resources for the command-line usage<a class="anchor" aria-label="anchor" href="#resources-for-the-command-line-usage"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<a href="usage.html">Code usage</a>: an example of the usage of
HaDeX functions.</li>
<li>
<a href="examples.html">Experimental examples</a>: a full analytic
workflow of HaDeX performed in the R console.</li>
</ol>
<pre><code>
 
# About HaDeX

HaDeX is a novel tool for processing, analysis, and visualization of HDX-MS experiments. HaDeX covers the final parts of the analytic process, including a comparison of experiments, quality control and generation of publication-quality figures. To make the HaDeX **R** package available to the less **R**-fluent users, we enhanced it with a comprehensive Graphical User Interface available as a HaDeX GUI. The reproducibility of the whole procedure is ensured with advanced reporting functions. 

The GUI is available online: http://mslab-ibb.pl/shiny/HaDeX/ or can be installed locally on Windows systems: https://sourceforge.net/projects/hadex/files/HaDeX_setup.exe/download. Alternatively, R-fluent users can access the GUI through the [HaDeX_gui()](https://hadexversum.github.io/HaDeX/reference/HaDeX_gui.html) function.  

This document covers the main functionalities of both the **R** package and the GUI.

## Comparison of the existing HDX-MS software

To show the novelty of HaDeX, we compare its functionalities with other relatively new software for analysis of HDX-MS data: MEMHDX [@HourdelMEMHDXinteractivetool2016] and Deuteros [@LauDeuterossoftwarerapid2019]. 



```{=html}
&lt;div class="datatables html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:100%;height:auto;"&gt;&lt;/div&gt;
&lt;script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3"&gt;{"x":{"style":"bootstrap","filter":"none","vertical":false,"data":[["Accessibility","Accessibility","Accessibility","Analysis","Analysis","Analysis","Analysis","Analysis","Analysis","Analysis","Analysis","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Reporting","Reporting","Reporting","Reporting","Reporting"],["Web server","Programmatic access","Desktop software","Multi-state analysis","ISO-based uncertainty","Coverage and peptide overlap","Kinetics analysis","Quality control","N- and C-terminal length corrections","Wald's test","Consolidation","Global visualization of deuterium uptake","Woods plot","Butterfly plot","Logit plot","Zooming of the Woods plot","Customizable label names and colors","Peptide kinetics chart","3D structure visualization","Downloadable charts","Deuterium uptake download","Downloadable results of intermediate computations","Report generation","PyMol export","HDX Data Summary"],["Yes","No","No","Yes","No","Yes","No","No","Yes","No","No","Yes","Yes","No","No","No","Yes","Yes","Yes","Yes","No","No","No","No","No"],["Yes","No","No","No","No","No","Yes","No","No","Yes","Yes","Yes","No","Yes","Yes","No","No","Yes","Yes","Yes","Yes","No","No","No","No"],["No","No","Yes","No","No","No","No","No","Yes","No","No","Yes","Yes","No","Yes","Yes","No","No","No","Yes","Yes","No","No","Yes","No"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","No","No","Yes","Yes","No","No","Yes","Yes","Yes","No","Yes","Yes","Yes","Yes","No","Yes"]],"container":"&lt;table class=\"table table-striped table-hover row-border order-column display\"&gt;\n  &lt;thead&gt;\n    &lt;tr&gt;\n      &lt;th&gt;Type&lt;\/th&gt;\n      &lt;th&gt;Name&lt;\/th&gt;\n      &lt;th&gt;MSTools&lt;\/th&gt;\n      &lt;th&gt;MEMHDX&lt;\/th&gt;\n      &lt;th&gt;Deuteros&lt;\/th&gt;\n      &lt;th&gt;HaDeX&lt;\/th&gt;\n    &lt;\/tr&gt;\n  &lt;\/thead&gt;\n&lt;\/table&gt;","options":{"dom":"t","ordering":false,"paging":false,"columnDefs":[{"name":"Type","targets":0},{"name":"Name","targets":1},{"name":"MSTools","targets":2},{"name":"MEMHDX","targets":3},{"name":"Deuteros","targets":4},{"name":"HaDeX","targets":5}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'background-color':value == \"Yes\" ? \"#00BFFF\" : value == \"No\" ? \"#FF8C91\" : null});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'background-color':value == \"Yes\" ? \"#00BFFF\" : value == \"No\" ? \"#FF8C91\" : null});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'background-color':value == \"Yes\" ? \"#00BFFF\" : value == \"No\" ? \"#FF8C91\" : null});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'background-color':value == \"Yes\" ? \"#00BFFF\" : value == \"No\" ? \"#FF8C91\" : null});\n}"},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.rowCallback"],"jsHooks":[]}&lt;/script&gt;</code></pre>
<p>We have not considered HDX Workbench <span class="citation">(Pascal
et al. 2012)</span> as it deals with the preliminary steps of the
analysis. This comparison also does not cover a versatile structural
visualisation tool for HDX-MS results, HDX-Viewer <span class="citation">(Bouyssié et al. 2019)</span> as its scope is different
from the tools mentioned above.</p>
<p><strong>Web server:</strong> a software is available as a web
server.</p>
<p><strong>Programmatic access:</strong> analytic functionalities are
documented and available from a command line.</p>
<p><strong>Desktop software:</strong> a software can be installed
locally.</p>
<p><strong>Multi-state analysis:</strong> a software supports
comparisons of more than two states.</p>
<p><strong>ISO-based uncertainty:</strong> analytic functions produce
ISO-compatible uncertainty intervals.</p>
<p><strong>Coverage and peptide overlap:</strong> overview of
experimental sequence coverage is available in a user-friendly way.</p>
<p><strong>Quality control:</strong> additional information about course
of the experiment.</p>
<p><strong>N- and C-terminal length corrections:</strong> manual
correction of sequence length.</p>
<p><strong>Global visualization of deuterium uptake:</strong> deuterium
uptake for different states is shown together for comparison.</p>
<p><strong>Woods plot:</strong> deuteration difference between chosen
states shown in the format of Woods plot.</p>
<p><strong>Zooming of the Woods plot:</strong> Woods plot can be zoomed
in.</p>
<p><strong>Customizable label names and colors:</strong> Labels and
colors on the plot can be changed by the user.</p>
<p><strong>Peptide kinetics chart:</strong> Kinetics plot (deuteration
change in time) are available for each peptide.</p>
<p><strong>3D structure visualization:</strong> structure of the protein
is visualized in 3D.</p>
<p><strong>Downloadable charts:</strong> charts are downloadable,
preferably in a vector format (.eps, .svg or .pdf).</p>
<p><strong>Deuterium uptake download:</strong> data shown on Woods plot
is downloadable (e.q. as CSV file).</p>
<p><strong>Downloadable results of intermediate computations:</strong>
results of intermediate computations (e.q. pure deuteration data) are
downloadable.</p>
<p><strong>Report generation:</strong> generates a report (e.q. in Html
format) with results of the analysis with parametrization.</p>
<p><strong>PyMol export:</strong> exports data to the <a href="https://pymol.org/2/" class="external-link">PyMol</a> format.</p>
<p><strong>HDX Data Summary:</strong> summary of the experimental data
<span class="citation">(Masson et al. 2019)</span>.</p>
</div>
<div class="section level2">
<h2 id="hadex-functionalities">HaDeX functionalities<a class="anchor" aria-label="anchor" href="#hadex-functionalities"></a>
</h2>
<div class="section level3">
<h3 id="data-import">Data import<a class="anchor" aria-label="anchor" href="#data-import"></a>
</h3>
<p>The HaDeX web server works only on data in the DynamX datafile format
(Waters Corp.). The data from other sources may be also adjusted to the
format accepted by HaDeX provided it has following columns:</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-e5c8c404fe174e4c81bd" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e5c8c404fe174e4c81bd">{"x":{"style":"bootstrap","filter":"none","vertical":false,"data":[["Protein","Start","End","Sequence","Modification","Fragment","MaxUptake","MHP","State","Exposure","File","z","RT","Inten","Center"],["Character","Integer","Integer","Character","Logic","Logic","Numeric","Numeric","Character","Numeric","Character","Integer","Numeric","Numeric","Numeric"]],"container":"<table class=\"table table-striped table-hover row-border order-column display\">\n  <thead>\n    <tr>\n      <th>Column.Name<\/th>\n      <th>Column.Type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","ordering":false,"paging":false,"autoWidth":true,"columnDefs":[{"name":"Column.Name","targets":0},{"name":"Column.Type","targets":1}],"order":[],"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script><p>Although data can be imported into R using other tools, we strongly
advise to rely on the <a href="https://hadexversum.github.io/HaDeX/reference/read_hdx.html" class="external-link">read_hdx()</a>
function:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># dat &lt;- HadeX::read_hdx(system.file(package = "HaDeX",            "HaDeX/data/KD_190304_Nucb2_EDTA_CaCl2_test02_clusterdata.csv"))</span></span></code></pre></div>
<p>Currently, <a href="https://hadexversum.github.io/HaDeX/reference/read_hdx.html" class="external-link">read_hdx()</a>
supports .csv, .tsv and .xls files fulfilling the data structure
described above. Files with data from multiple proteins are supported.
The user need to indicate which protein is of interest for
calculations.</p>
</div>
<div class="section level3">
<h3 id="computation-of-deuteration-levels">Computation of deuteration levels<a class="anchor" aria-label="anchor" href="#computation-of-deuteration-levels"></a>
</h3>
<p>The computation of the level of deuteration involves several
pre-processing steps, all of which are described in this section. These
steps are performed automatically in the GUI or by the <a href="https://hadexversum.github.io/HaDeX/reference/prepare_dataset.html" class="external-link">prepare_dataset()</a>
function in the console.</p>
<div class="section level4">
<h4 id="measured-data-into-overall-peptide-mass">Measured data into overall peptide mass<a class="anchor" aria-label="anchor" href="#measured-data-into-overall-peptide-mass"></a>
</h4>
<p>The results of HDX-MS measurements as given in the DynamX data files
are represented as the measured mass of peptides plus proton mass to
charge ratio
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>C</mi><mi>e</mi><mi>n</mi><mi>t</mi><mi>e</mi><mi>r</mi></mrow><annotation encoding="application/x-tex">Center</annotation></semantics></math>).
For later use, this value has to be transformed into an overall mass of
a peptide measured after specific time point from a protein in a
specific state, as shown in equation 1:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mi>e</mi><mi>p</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi><mo>=</mo><mi>z</mi><mo>×</mo><mrow><mo stretchy="true" form="prefix">(</mo><mi>C</mi><mi>e</mi><mi>n</mi><mi>t</mi><mi>e</mi><mi>r</mi><mo>−</mo><mi>p</mi><mi>r</mi><mi>o</mi><mi>t</mi><mi>o</mi><mi>n</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">pepMass = z \times (Center-protonMass)\tag{1}</annotation></semantics></math>
where:</p>
<ul>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mi>e</mi><mi>p</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi></mrow><annotation encoding="application/x-tex">pepMass</annotation></semantics></math>
- expected mass of the peptide after incubation (Da),</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mi>r</mi><mi>o</mi><mi>t</mi><mi>o</mi><mi>n</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi></mrow><annotation encoding="application/x-tex">protonMass</annotation></semantics></math>
- the mass of the proton (Da),</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>z</mi><annotation encoding="application/x-tex">z</annotation></semantics></math>
- charge of the peptide,</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>C</mi><mi>e</mi><mi>n</mi><mi>t</mi><mi>e</mi><mi>r</mi></mrow><annotation encoding="application/x-tex">Center</annotation></semantics></math>
- experimentally measured peptide mass plus proton mass to charge ratio
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><mfrac><mi>m</mi><mi>z</mi></mfrac><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">\left(\frac{m}{z}\right)</annotation></semantics></math>.</p></li>
</ul>
<p>HDX-MS experiments are often repeated (by the rule of thumb at least
three times). Thus, we aggregate the results of replicates as a weighted
mean mass into a single result per peptide using equation 2:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>a</mi><mi>g</mi><mi>g</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi><mo>=</mo><munderover><mo>∑</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>N</mi></munderover><mfrac><mrow><mi>I</mi><mi>n</mi><mi>t</mi><mi>e</mi><msub><mi>n</mi><mi>k</mi></msub></mrow><mi>N</mi></mfrac><mo>×</mo><mi>p</mi><mi>e</mi><mi>p</mi><mi>M</mi><mi>a</mi><mi>s</mi><msub><mi>s</mi><mi>k</mi></msub></mrow><annotation encoding="application/x-tex">aggMass = \sum_{k = 1}^{N}\frac{Inten_k}{N}{\times}pepMass_k\tag{2}</annotation></semantics></math>
where:</p>
<ul>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>a</mi><mi>g</mi><mi>g</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi></mrow><annotation encoding="application/x-tex">aggMass</annotation></semantics></math>
- weighted mean mass of the peptide (Da),</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
- replicate index,</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>I</mi><mi>n</mi><mi>t</mi><mi>e</mi><mi>n</mi></mrow><annotation encoding="application/x-tex">Inten</annotation></semantics></math>
- intensity,</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>N</mi><annotation encoding="application/x-tex">N</annotation></semantics></math>
- number of replicates.</p></li>
</ul>
<p>This data manipulation results from an original data structure. Each
repetition of the measurement gives the data for given peptide in a
given time per possible value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>z</mi><annotation encoding="application/x-tex">z</annotation></semantics></math>
as shown in the example below. We need to use the value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mi>e</mi><mi>p</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi></mrow><annotation encoding="application/x-tex">pepMass</annotation></semantics></math>
as shown in equation 1 but still keep the information about original
measurement - that’s why we use the weighted mass.</p>
<p>The uncertainty of a measurement is variability associated with the
precision of measuring instrumentation. We present here a novel
derivation of uncertainty formulas for HDX-MS data according to the ISO
guidelines <span class="citation">(Joint Committee for Guides in
Metrology 2008)</span>. Input files always encompass results of more
than one measurement. We assume uncorrelatedness of replicates as they
come from different samples. Therefore, we average measurements of
replicates for each time point and for all protein states. Thus, we
compute peptide mass uncertainty
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>u</mi><annotation encoding="application/x-tex">u</annotation></semantics></math>
as uncertainty for aggregate estimate using the formula for standard
deviation of the mean:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><msqrt><mfrac><mrow><munderover><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></munderover><msup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mi>i</mi></msub><mo>−</mo><mover><mi>x</mi><mo accent="true">¯</mo></mover><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mrow><mrow><mi>n</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>n</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac></msqrt></mrow><annotation encoding="application/x-tex">u(x) = \sqrt{\frac{ \sum_{i=1}^n \left( x_{i} - \overline{x} \right)^2}{n(n-1)}}\tag{3}</annotation></semantics></math>
where:</p>
<ul>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>x</mi><mi>i</mi></msub><annotation encoding="application/x-tex">x_{i}</annotation></semantics></math>
- measurement,</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mover><mi>x</mi><mo accent="true">¯</mo></mover><annotation encoding="application/x-tex">\overline{x}</annotation></semantics></math>
- mean value,</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>n</mi><annotation encoding="application/x-tex">n</annotation></semantics></math>
- number of measurements.</p></li>
</ul>
<p>After obtaining the mass of the peptide, we can compute the
deuteration level depending on the chosen maximum deuteration level. The
maximum deuteration can also be computed in two different ways: either
as <em>theoretical</em> (where the maximum deuteration depends on the
theoretical deuteration levels) and <em>experimental</em> (where the
maximum deuteration is assumed to be equal to the deuteration measured
at the last time point).</p>
</div>
<div class="section level4">
<h4 id="experimental-deuteration-level">Experimental deuteration level<a class="anchor" aria-label="anchor" href="#experimental-deuteration-level"></a>
</h4>
<p>The experimental deuteration level is computed as the deuteration
level of the peptide from a protein in a specific state and after
incubation time
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>t</mi><annotation encoding="application/x-tex">t</annotation></semantics></math>
compared to the deuteration level measured at the start of the
incubation
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>t</mi><mn>0</mn></msub><annotation encoding="application/x-tex">t_0</annotation></semantics></math>).
It yields a value for the chosen state and chosen time
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>t</mi><annotation encoding="application/x-tex">t</annotation></semantics></math>.</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>D</mi><mo>=</mo><msub><mi>D</mi><mi>t</mi></msub><mo>−</mo><msub><mi>D</mi><msub><mi>t</mi><mn>0</mn></msub></msub></mrow><annotation encoding="application/x-tex">D = D_{t} - D_{t_0}\tag{4}</annotation></semantics></math></p>
<p>where:</p>
<ul>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>D</mi><annotation encoding="application/x-tex">D</annotation></semantics></math>
- deuteration level (Da),</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>D</mi><msub><mi>t</mi><mn>0</mn></msub></msub><annotation encoding="application/x-tex">D_{t_0}</annotation></semantics></math>
- experimentally measured deuteration at the beginning of the incubation
(0 or close to 0),</p></li>
</ul>
<p>The equation 4 produces only absolute deuteration levels. The
computations of relative deuteration levels follows a similar logic and
is normalized by the difference of deuteration between the start
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>t</mi><mn>0</mn></msub><annotation encoding="application/x-tex">t_0</annotation></semantics></math>)
and the end of the experiment
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>t</mi><mi>o</mi></msub><mi>u</mi><mi>t</mi></mrow><annotation encoding="application/x-tex">t_out</annotation></semantics></math>)
as shown in the equation 4a:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>D</mi><mo>=</mo><mfrac><mrow><msub><mi>D</mi><mi>t</mi></msub><mo>−</mo><msub><mi>D</mi><msub><mi>t</mi><mn>0</mn></msub></msub></mrow><mrow><msub><mi>D</mi><msub><mi>t</mi><mrow><mi>o</mi><mi>u</mi><mi>t</mi></mrow></msub></msub><mo>−</mo><msub><mi>D</mi><msub><mi>t</mi><mn>0</mn></msub></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex">D = \frac{D_{t} - D_{t_0}}{D_{t_{out}} - D_{t_0}}\tag{4a}</annotation></semantics></math></p>
<p>All functions in the HaDeX package contain the logical parameter
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>r</mi><mi>e</mi><mi>l</mi><mi>a</mi><mi>t</mi><mi>i</mi><mi>v</mi><mi>e</mi></mrow><annotation encoding="application/x-tex">relative</annotation></semantics></math>
to determine if they should return absolute or relative deuteration
levels.</p>
<div class="section level5">
<h5 id="uncertainty-calculations">Uncertainty calculations<a class="anchor" aria-label="anchor" href="#uncertainty-calculations"></a>
</h5>
<p>We describe the methodology of the uncertainty calculations for
relative deuteration levels. The uncertainty for absolute deuteration
levels is computed similarly, but without scaling.</p>
<p>To calculate uncertainty related to functions of more than one
variables (e. g., equation 4) the Law of propagation of uncertainty is
defined by equation 5:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>u</mi><mi>c</mi></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>y</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><msqrt><mrow><munder><mo>∑</mo><mi>k</mi></munder><msup><mrow><mo stretchy="true" form="prefix">[</mo><mfrac><mrow><mi>∂</mi><mi>y</mi></mrow><mrow><mi>∂</mi><msub><mi>x</mi><mi>k</mi></msub></mrow></mfrac><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><mn>2</mn></msup></mrow></msqrt></mrow><annotation encoding="application/x-tex">u_{c}(y) = \sqrt{\sum_{k} \left[ \frac{\partial y}{\partial x_{k}} u(x_{k}) \right]^2}\tag{5}</annotation></semantics></math></p>
<p>As the variable of interest is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>D</mi><annotation encoding="application/x-tex">D</annotation></semantics></math>,
we apply the general formula to the deuteration level
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>D</mi><annotation encoding="application/x-tex">D</annotation></semantics></math>
(equation 6):</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>u</mi><mi>c</mi></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>D</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><msqrt><mrow><munder><mo>∑</mo><mi>k</mi></munder><msup><mrow><mo stretchy="true" form="prefix">[</mo><mfrac><mrow><mi>∂</mi><mi>D</mi></mrow><mrow><mi>∂</mi><msub><mi>D</mi><mi>k</mi></msub></mrow></mfrac><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><mn>2</mn></msup></mrow></msqrt></mrow><annotation encoding="application/x-tex">u_{c}(D) = \sqrt{\sum_{k} \left[ \frac{\partial D}{\partial D_{k}} u(D_{k}) \right]^2 }\tag{6}</annotation></semantics></math></p>
<p>where:</p>
<ul>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>0</mn><mo>,</mo><mi>t</mi><mo>,</mo><mi>o</mi><mi>u</mi><mi>t</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">k \in \{0, t, out\}</annotation></semantics></math>,</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>D</mi><mi>k</mi></msub><annotation encoding="application/x-tex">D_{k}</annotation></semantics></math>
- deuteration in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
time (Da),</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">u(D_{k})</annotation></semantics></math>
- an uncertainty associated with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>D</mi><mi>k</mi></msub><annotation encoding="application/x-tex">D_{k}</annotation></semantics></math>
as standard deviation of the mean value,</p></li>
</ul>
<p>Then, expanding the equation 6:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>u</mi><mi>c</mi></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>D</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><msqrt><mrow><msup><mrow><mo stretchy="true" form="prefix">[</mo><mfrac><mn>1</mn><mrow><msub><mi>D</mi><msub><mi>t</mi><mrow><mi>o</mi><mi>u</mi><mi>t</mi></mrow></msub></msub><mo>−</mo><msub><mi>D</mi><msub><mi>t</mi><mn>0</mn></msub></msub></mrow></mfrac><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mi>t</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><mn>2</mn></msup><mo>+</mo><msup><mrow><mo stretchy="true" form="prefix">[</mo><mfrac><mrow><msub><mi>D</mi><mi>t</mi></msub><mo>−</mo><msub><mi>D</mi><msub><mi>t</mi><mrow><mi>o</mi><mi>u</mi><mi>t</mi></mrow></msub></msub></mrow><msup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><msub><mi>t</mi><mrow><mi>o</mi><mi>u</mi><mi>t</mi></mrow></msub></msub><mo>−</mo><msub><mi>D</mi><msub><mi>t</mi><mn>0</mn></msub></msub><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mfrac><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><msub><mi>t</mi><mn>0</mn></msub></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><mn>2</mn></msup><mo>+</mo><msup><mrow><mo stretchy="true" form="prefix">[</mo><mfrac><mrow><msub><mi>D</mi><msub><mi>t</mi><mn>0</mn></msub></msub><mo>−</mo><msub><mi>D</mi><mi>t</mi></msub></mrow><msup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><msub><mi>t</mi><mrow><mi>o</mi><mi>u</mi><mi>t</mi></mrow></msub></msub><mo>−</mo><msub><mi>D</mi><msub><mi>t</mi><mn>0</mn></msub></msub><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mfrac><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><msub><mi>t</mi><mrow><mi>o</mi><mi>u</mi><mi>t</mi></mrow></msub></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><mn>2</mn></msup></mrow></msqrt></mrow><annotation encoding="application/x-tex">u_{c}(D) = \sqrt{ \left[ 
\frac{1}{D_{t_{out}}-D_{t_0}} u(D_{t}) \right]^2 + 
\left[ \frac{D_{t} - D_{t_{out}}}{(D_{t_{out}}-D_{t_0})^2} u(D_{t_0}) \right]^2 + 
\left[ \frac{D_{t_0} - D_{t}}{(D_{t_{out}}-D_{t_0})^2} u (D_{t_{out}}) \right]^2}\tag{7}</annotation></semantics></math>
As expected, the uncertainty associated with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>D</mi><mi>t</mi></msub><annotation encoding="application/x-tex">D_{t}</annotation></semantics></math>
has the biggest impact on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>u</mi><mi>c</mi></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>D</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">u_{c}(D)</annotation></semantics></math>.</p>
</div>
</div>
<div class="section level4">
<h4 id="theoretical-deuteration-level">Theoretical deuteration level<a class="anchor" aria-label="anchor" href="#theoretical-deuteration-level"></a>
</h4>
<p>As opposed to the experimental deuteration levels, theoretical
deuteration level only partially depends on the experimental data. Here,
the maximum deuteration level is based on a hypothetical peptide where
all hydrogens were replaced by deuterons, as it is shown in equation
8:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>D</mi><mo>=</mo><mfrac><mrow><msub><mi>D</mi><mi>t</mi></msub><mo>−</mo><mi>M</mi><mi>H</mi><mi>P</mi></mrow><mrow><mi>M</mi><mi>a</mi><mi>x</mi><mi>U</mi><mi>p</mi><mi>t</mi><mi>a</mi><mi>k</mi><mi>e</mi><mo>×</mo><mi>p</mi><mi>r</mi><mi>o</mi><mi>t</mi><mi>o</mi><mi>n</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi></mrow></mfrac></mrow><annotation encoding="application/x-tex">D = \frac{D_{t}-MHP}{MaxUptake \times protonMass}\tag{8}</annotation></semantics></math></p>
<p>where:</p>
<ul>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>D</mi><mi>t</mi></msub><annotation encoding="application/x-tex">D_{t}</annotation></semantics></math>
- deuteration measured in a chosen time point (Da),</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mi>H</mi><mi>P</mi></mrow><annotation encoding="application/x-tex">MHP</annotation></semantics></math>
- theoretical mass of the peptide (constant) (Da),</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mi>a</mi><mi>x</mi><mi>U</mi><mi>p</mi><mi>t</mi><mi>a</mi><mi>k</mi><mi>e</mi></mrow><annotation encoding="application/x-tex">MaxUptake</annotation></semantics></math>
- the maximum proton uptake for the peptide (theoretical constant)
(Da),</p></li>
<li><p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mi>r</mi><mi>o</mi><mi>t</mi><mi>o</mi><mi>n</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi></mrow><annotation encoding="application/x-tex">protonMass</annotation></semantics></math>
- mass of a proton (constant) (Da).</p></li>
</ul>
<p>The absolute deuteration level is calculated as in equation 8 but
without scaling (equation 8a):</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>D</mi><mo>=</mo><msub><mi>D</mi><mi>t</mi></msub><mo>−</mo><mi>M</mi><mi>H</mi><mi>P</mi></mrow><annotation encoding="application/x-tex">D = D_{t} - MHP\tag{8a}</annotation></semantics></math></p>
<div class="section level5">
<h5 id="uncertainty-calculations-1">Uncertainty calculations<a class="anchor" aria-label="anchor" href="#uncertainty-calculations-1"></a>
</h5>
<p>For functions of one variable uncertainty reduces to:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>y</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mrow><mo stretchy="true" form="prefix">|</mo><mfrac><mrow><mi>d</mi><mi>y</mi></mrow><mrow><mi>d</mi><mi>x</mi></mrow></mfrac><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">|</mo></mrow><mi>.</mi></mrow><annotation encoding="application/x-tex">u(y) = \left| \frac{dy}{dx} u(x) \right|.\tag{9}</annotation></semantics></math></p>
<p>Substituting
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>D</mi><annotation encoding="application/x-tex">D</annotation></semantics></math>
from equation 8, we have</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>D</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mrow><mo stretchy="true" form="prefix">|</mo><mfrac><mn>1</mn><mrow><mi>M</mi><mi>a</mi><mi>x</mi><mi>U</mi><mi>p</mi><mi>t</mi><mi>a</mi><mi>k</mi><mi>e</mi><mo>×</mo><mi>p</mi><mi>r</mi><mi>o</mi><mi>t</mi><mi>o</mi><mi>n</mi><mi>M</mi><mi>a</mi><mi>s</mi><mi>s</mi></mrow></mfrac><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mi>t</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">|</mo></mrow></mrow><annotation encoding="application/x-tex">u(D) = \left|\frac{1}{MaxUptake \times protonMass} u(D_{t}) \right|\tag{10}</annotation></semantics></math></p>
<p>For the absolute values,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>D</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">u(D)</annotation></semantics></math>
is identical with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>u</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mi>t</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">u(D_{t})</annotation></semantics></math>,
based on equations 8a and 9.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="difference-of-deuteration-levels-between-two-states">Difference of deuteration levels between two states<a class="anchor" aria-label="anchor" href="#difference-of-deuteration-levels-between-two-states"></a>
</h3>
<p>The differences of deuteration levels between two states are
associated with a different level of protection of hydrogens. Therefore,
we are especially interested in the differential analysis of the
deuteration levels. Thus, the deuteration level in one state
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mn>2</mn></msub><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(D_{2})</annotation></semantics></math>
is subtracted from deuteration level in the other state
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mn>1</mn></msub><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(D_{1})</annotation></semantics></math>:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>d</mi><mi>i</mi><mi>f</mi><mi>f</mi><mo>=</mo><msub><mi>D</mi><mn>1</mn></msub><mo>−</mo><msub><mi>D</mi><mn>2</mn></msub></mrow><annotation encoding="application/x-tex">diff = D_{1} - D_{2}\tag{11}</annotation></semantics></math></p>
<p>and the uncertainty is a function of two variables (based on equation
11 and 5):</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>u</mi><mi>c</mi></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>d</mi><mi>i</mi><mi>f</mi><mi>f</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><msqrt><mrow><mi>u</mi><msup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mn>1</mn></msub><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mo>+</mo><mi>u</mi><msup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mn>2</mn></msub><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mrow></msqrt></mrow><annotation encoding="application/x-tex">u_{c}(diff) = \sqrt{u(D_{1})^2 + u(D_{2})^2}\tag{12}</annotation></semantics></math></p>
</div>
<div class="section level3">
<h3 id="visual-data-analysis">Visual data analysis<a class="anchor" aria-label="anchor" href="#visual-data-analysis"></a>
</h3>
<div class="section level4">
<h4 id="comparison-of-states">Comparison of states<a class="anchor" aria-label="anchor" href="#comparison-of-states"></a>
</h4>
<p>Comparison plots show the deuteration level of all peptides in
selected states in a given time. The x-axis represents positions of
amino acids in the sequence. The y-axis shows the deuteration level,
expressed either as a <em>relative</em> or <em>absolute</em>
deuteration. The chart below shows peptide deuteration after 25 minutes
of the incubation along with the confidence intervals.</p>
<p>– theoretical:</p>
<ul>
<li>relative values:</li>
</ul>
<p><!-- ```{r warning=FALSE} -->
<!-- comparison_plot(calc_dat = calc_dat, -->
<!--                 theoretical = TRUE, -->
<!--                 relative = TRUE, -->
<!--                 state_first = "Nucb2 Factor 1", -->
<!--                 state_second = "Nucb2 Factor 2") + -->
<!--   labs(title = "Theoretical fraction exchanged in state comparison in 25 min time") -->
<!-- ``` --></p>
<ul>
<li>absolute values:</li>
</ul>
<p><!-- ```{r warning=FALSE} -->
<!-- comparison_plot(calc_dat = calc_dat, -->
<!--                 theoretical = TRUE, -->
<!--                 relative = FALSE, -->
<!--                 state_first = "Nucb2 Factor 1", -->
<!--                 state_second = "Nucb2 Factor 2") + -->
<!--   labs(title = "Theoretical fraction exchanged in state comparison in 25 min time") -->
<!-- ``` --></p>
<p>– experimental:</p>
<ul>
<li>relative values:</li>
</ul>
<p><!-- ```{r warning=FALSE} -->
<!-- comparison_plot(calc_dat = calc_dat, -->
<!--                 theoretical = FALSE, -->
<!--                 relative = TRUE,  -->
<!--                 state_first = "Nucb2 Factor 1", -->
<!--                 state_second = "Nucb2 Factor 2") + -->
<!--   labs(title = "Fraction exchanged in state comparison in 25 min time") -->
<!-- ``` --></p>
<ul>
<li>absolute values:</li>
</ul>
<p><!-- ```{r warning=FALSE} -->
<!-- comparison_plot(calc_dat = calc_dat, -->
<!--                 theoretical = FALSE, -->
<!--                 relative = FALSE,  -->
<!--                 state_first = "Nucb2 Factor 1", -->
<!--                 state_second = "Nucb2 Factor 2") + -->
<!--   labs(title = "Fraction exchanged in state comparison in 25 min time") -->
<!-- ``` --></p>
</div>
</div>
<div class="section level3">
<h3 id="woods-plot">Woods plot<a class="anchor" aria-label="anchor" href="#woods-plot"></a>
</h3>
<p>Woods plots show the difference between the deuteration of all
peptides in two different states in a specific time point as described
by equation 11. Similarly to the comparison plot, HaDeX provides both
experimental and theoretical deuteration levels using either relative or
absolute values:</p>
<p>– theoretical:</p>
<ul>
<li>relative values:</li>
</ul>
<p><!-- ```{r warning=FALSE} -->
<!-- woods_plot(calc_dat = calc_dat, -->
<!--            theoretical = TRUE, -->
<!--            relative = TRUE) + -->
<!--   labs(title = "Theoretical fraction exchanged between states in 25 min time") -->
<!-- ``` --></p>
<ul>
<li>absolute values:</li>
</ul>
<p><!-- ```{r warning=FALSE} -->
<!-- woods_plot(calc_dat = calc_dat, -->
<!--            theoretical = TRUE, -->
<!--            relative = FALSE) + -->
<!--   labs(title = "Theoretical fraction exchanged between states in 25 min time") -->
<!-- ``` --></p>
<p>– experimental:</p>
<ul>
<li>relative values:</li>
</ul>
<p><!-- ```{r warning=FALSE } -->
<!-- woods_plot(calc_dat = calc_dat, -->
<!--            theoretical = FALSE,  -->
<!--            relative = TRUE) + -->
<!--   labs(title = "Theoretical fraction exchanged between states in 25 min time") -->
<!-- ``` --></p>
<ul>
<li>absolute values:</li>
</ul>
<p><!-- ```{r warning=FALSE } -->
<!-- woods_plot(calc_dat = calc_dat, -->
<!--            theoretical = FALSE,  -->
<!--            relative = FALSE) + -->
<!--   labs(title = "Theoretical fraction exchanged between states in 25 min time") -->
<!-- ``` --></p>
<div class="section level4">
<h4 id="confidence-limit-in-woods-plot">Confidence limit in Woods plot<a class="anchor" aria-label="anchor" href="#confidence-limit-in-woods-plot"></a>
</h4>
<p>The function <a href="https://hadexversum.github.io/HaDeX/reference/calculate_confidence_limit_values.html" class="external-link">calculate_confidence_limit_values()</a>
calculates confidence limit values as it is described elsewhere <span class="citation">(Houde, Berkowitz, and Engen 2011)</span>.</p>
<p><!-- ```{r} -->
<!-- calculate_confidence_limit_values(calc_dat = calc_dat, -->
<!--                                   confidence_limit = 0.99, -->
<!--                                   theoretical = FALSE,  -->
<!--                                   relative = TRUE)   -->
<!-- ```  --></p>
<p>The function <a href="https://hadexversum.github.io/HaDeX/reference/add_stat_dependency.html" class="external-link">add_stat_dependency()</a>
returns data extended by column describing relation of a given peptide
with confidence limit.</p>
<p><!-- ```{r} --> <!-- add_stat_dependency(calc_dat,  -->
<!--                    confidence_limit = 0.98,  -->
<!--                    theoretical = FALSE,  -->
<!--                    relative = TRUE) --> <!-- ``` --></p>
</div>
</div>
<div class="section level3">
<h3 id="kinetic-plots">Kinetic plots<a class="anchor" aria-label="anchor" href="#kinetic-plots"></a>
</h3>
<p>By the term <em>kinetics</em> we understand deuteration change in
time. To calculate deuteration values per peptide for different time
points is used function <a href="https://hadexversum.github.io/HaDeX/reference/calculate_kinetics.html" class="external-link">calculate_kinetics()</a>.
This function uses the <a href="https://hadexversum.github.io/HaDeX/reference/calculate_state_deuteration.html" class="external-link">calculate_state_deuteration()</a>
function.</p>
<!-- ```{r warning = FALSE} -->
<!-- (kin_YYDEYL_gg_Nucb2_CaCl2 <- calculate_kinetics(dat = dat,  -->
<!--                                                 protein = "db_Nucb2",  -->
<!--                                                 sequence = "YYDEYL", -->
<!--                                                 state = "gg_Nucb2_CaCl2",  -->
<!--                                                 start = 45,  -->
<!--                                                 end = 50,  -->
<!--                                                 time_in = 0.001,  -->
<!--                                                 time_out = 1440)) -->
<!-- ``` -->
<!-- ```{r warning = FALSE} -->
<!-- (kin_YYDEYL_gg_Nucb2_EDTA <- calculate_kinetics(dat = dat,  -->
<!--                                               protein = "db_Nucb2",  -->
<!--                                               sequence = "YYDEYL", -->
<!--                                               state = "gg_Nucb2_EDTA",  -->
<!--                                               start = 45,  -->
<!--                                               end = 50,  -->
<!--                                               time_in = 0.001,  -->
<!--                                               time_out = 1440)) -->
<!-- ``` -->
<p>Calculated data can be shown next to each other on the plot for
comparison. To visualize kinetic data we recommend <a href="https://hadexversum.github.io/HaDeX/reference/plot_kinetics.html" class="external-link">plot_kinetics()</a>
function:</p>
<p>– theoretical:</p>
<ul>
<li>relative values:</li>
</ul>
<!-- ```{r warning = FALSE} --><!-- bind_rows(kin_YYDEYL_gg_Nucb2_CaCl2, kin_YYDEYL_gg_Nucb2_EDTA) %>% --><!--   plot_kinetics(theoretical = TRUE,  --><!--                 relative = TRUE) --><!-- ``` --><ul>
<li>absolute values:</li>
</ul>
<!-- ```{r warning = FALSE} --><!-- bind_rows(kin_YYDEYL_gg_Nucb2_CaCl2, kin_YYDEYL_gg_Nucb2_EDTA) %>% --><!--   plot_kinetics(theoretical = TRUE,  --><!--                 relative = FALSE) --><!-- ``` --><p>– experimental:</p>
<ul>
<li>relative values:</li>
</ul>
<!-- ```{r warning = FALSE} --><!-- bind_rows(kin_YYDEYL_gg_Nucb2_CaCl2, kin_YYDEYL_gg_Nucb2_EDTA) %>% --><!--   plot_kinetics(theoretical = FALSE,  --><!--                 relative = TRUE) --><!-- ``` --><ul>
<li>absolute values:</li>
</ul>
<!-- ```{r warning = FALSE} --><!-- bind_rows(kin_YYDEYL_gg_Nucb2_CaCl2, kin_YYDEYL_gg_Nucb2_EDTA) %>% --><!--   plot_kinetics(theoretical = FALSE,  --><!--                 relative = FALSE) --><!-- ``` -->
</div>
<div class="section level3">
<h3 id="additional-tools">Additional tools<a class="anchor" aria-label="anchor" href="#additional-tools"></a>
</h3>
<p>HaDeX provides additional tools for assessment of experiments.</p>
<div class="section level4">
<h4 id="peptide-coverage">Peptide coverage<a class="anchor" aria-label="anchor" href="#peptide-coverage"></a>
</h4>
<p>The sequence of the protein(s) is reconstructed from the peptides
from the input file. Thus, amino acids not covered by peptides are
marked as X according to the IUPAC convention. The sequence is
reconstructed using the <a href="https://hadexversum.github.io/HaDeX/reference/reconstruct_sequence.html" class="external-link">reconstruct_sequence()</a>
function.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># reconstruct_sequence(dat)</span></span></code></pre></div>
<p>Additionally, the coverage of peptides can be presented on a chart
using the <a href="https://hadexversum.github.io/HaDeX/reference/plot_coverage.html" class="external-link">plot_coverage()</a>
and <a href="https://hadexversum.github.io/HaDeX/reference/plot_position_frequency.html" class="external-link">plot_position_frequency()</a>
functions.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot_coverage(dat, chosen_state = "gg_Nucb2_CaCl2")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># plot_position_frequency(dat, chosen_state = "gg_Nucb2_CaCl2")</span></span></code></pre></div>
<p>The user can choose which state (or states) should be included in
these plots. If this parameter is not provided, the first possible state
is chosen. If a given peptide is available in more than one state, it is
shown only once.</p>
</div>
<div class="section level4">
<h4 id="quality-control">Quality control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h4>
<p>The function <a href="https://hadexversum.github.io/HaDeX/reference/quality_control.html" class="external-link">quality_control()</a>
plots the change in the uncertainty of deuteration levels as a function
of incubation time. The uncertainty is averaged over all peptides
available at a given time point in a selected state. This chart has a
double function: firstly, it allows checking if the measurement
uncertainty is decreasing over time (which is the expected behavior) and
the second one helps to plan the appropriate incubation length for the
tested protein (whether we obtain the desired data reliability
values).</p>
<p>In HDX-MS experiments, the average uncertainty of measurements
decreases over time as the sample is more and more deuterated (even
accounting for the back-propagation). Therefore, if the level of
uncertainty is stable, the exchange is still ongoing and it would be
advisable to prolong the reaction.</p>
<p>Moreover, the user can detect a time point after which the decrease
of the deuteration uncertainty becomes too marginal to prolong the
measurements. This function is most useful in case of multiple
measurements of the same or very similar proteins because it helps to
optimize the duration of the incubation. The result of this function can
be easily visualized.</p>
<p>Example:</p>
<p><!-- ```{r warning=FALSE} -->
<!-- result <- quality_control(dat = dat, -->
<!--                           state_first = "gg_Nucb2_EDTA", -->
<!--                           state_second = "gg_Nucb2_CaCl2",  -->
<!--                           chosen_time = 1,  -->
<!--                           in_time = 0.001) --></p>
<p><!-- ``` --></p>
<p><!-- ```{r warning=FALSE} --> <!-- ggplot(result) +  -->
<!--   geom_line(aes(x = out_time, y = avg_err_state_first, color = "Average error (first state)")) + -->
<!--   geom_line(aes(x = out_time, y = avg_err_state_second, color = "Average error (second state)")) + -->
<!--   scale_x_log10() + -->
<!--   labs(x = "log(time) [min]", y = "Average uncertainty", title = "Uncertainty change") +  -->
<!--   theme_bw(base_size = 11) + -->
<!--   theme(legend.position = "bottom", -->
<!--         legend.title = element_blank()) --> <!-- ``` --></p>
<p>The chart below shows how we interpret the three most common results
of quality control.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="hadex-graphical-user-interface">HaDeX Graphical User Interface<a class="anchor" aria-label="anchor" href="#hadex-graphical-user-interface"></a>
</h2>
<p>The HaDeX Shiny app is launched by the <a href="https://hadexversum.github.io/HaDeX/reference/HaDeX_gui.html" class="external-link">HaDeX_gui()</a>
function or available at MS Lab website: <a href="http://mslab-ibb.pl/shiny/HaDeX/" class="external-link uri">http://mslab-ibb.pl/shiny/HaDeX/</a>. In the Input Data tab
the user can adjust parameters to be propagated for the whole
analysis.</p>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<div class="section level3">
<h3 id="example-1-cd160-hvem">Example 1: CD160-HVEM<a class="anchor" aria-label="anchor" href="#example-1-cd160-hvem"></a>
</h3>
<p>The interaction between HVEM and the <a href="https://www.uniprot.org/uniprot/O95971" class="external-link">CD160</a> receptor was
measured with HDX-MS.</p>
<p>Firstly, we read the input data, exactly as provided by the DynamX
3.0 (Waters Corp.)</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://hadexversum.github.io/HaDeX2/">HaDeX</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># file import</span></span>
<span><span class="va">dat_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_hdx.html">read_hdx</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"HaDeX"</span>, </span>
<span>                              <span class="st">"HaDeX/data/KD_180110_CD160_HVEM.csv"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] ""</span></span></code></pre>
<p>Then, we reconstruct the protein sequence from the peptides measured
during the experiment. We observe the region from amino acid 107 till
amino acid 124 is not covered by any peptide.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/reconstruct_sequence.html">reconstruct_sequence</a></span><span class="op">(</span><span class="va">dat_1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_position_frequency.html">plot_position_frequency</a></span><span class="op">(</span><span class="va">dat_1</span>, chosen_state <span class="op">=</span> <span class="st">"CD160"</span><span class="op">)</span></span></code></pre></div>
<p>The theoretical plot allows finding regions, which exchange quickly
(N-terminus, regions between 30-70 amino acid) and regions, which
exchange slowly (peptides 15-24, 95-110 amino acid). We can also see the
differences in the exchange between two states, indicating regions which
changed upon binding with other protein.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate data</span></span>
<span><span class="co"># calc_dat_1 &lt;- prepare_dataset(dat = dat_1,</span></span>
<span><span class="co">#                               in_state_first = "CD160_0.001",</span></span>
<span><span class="co">#                               chosen_state_first = "CD160_1",</span></span>
<span><span class="co">#                               out_state_first = "CD160_1440",</span></span>
<span><span class="co">#                               in_state_second = "CD160_HVEM_0.001",</span></span>
<span><span class="co">#                               chosen_state_second = "CD160_HVEM_1",</span></span>
<span><span class="co">#                               out_state_second = "CD160_HVEM_1440")                             </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # theoretical comparison plot - relative values</span></span>
<span><span class="co"># comparison_plot(calc_dat = calc_dat_1,</span></span>
<span><span class="co">#                 theoretical = TRUE,</span></span>
<span><span class="co">#                 relative = TRUE, </span></span>
<span><span class="co">#                 state_first = "CD160",</span></span>
<span><span class="co">#                 state_second = "CD160_HVEM")</span></span></code></pre></div>
<p>On the experimental plot, there are visible regions, which exchange
quickly (N terminal part, regions between 30-70 amino acid) and regions,
which exchange slowly (peptides 15-24, 30-35, 95-110 amino acid).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># experimental comparison plot - relative values</span></span>
<span><span class="co"># comparison_plot(calc_dat = calc_dat_1,</span></span>
<span><span class="co">#                 theoretical = FALSE,</span></span>
<span><span class="co">#                 relative = TRUE,</span></span>
<span><span class="co">#                 state_first = "CD160",</span></span>
<span><span class="co">#                 state_second = "CD160_HVEM")</span></span></code></pre></div>
<p>Plots below are equivalent to plots above but in absolute values -
for users that prefer those.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># theoretical comparison plot - absolute values</span></span>
<span><span class="co"># comparison_plot(calc_dat = calc_dat_1,</span></span>
<span><span class="co">#                 theoretical = TRUE,</span></span>
<span><span class="co">#                 relative = FALSE, </span></span>
<span><span class="co">#                 state_first = "CD160",</span></span>
<span><span class="co">#                 state_second = "CD160_HVEM")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # experimental comparison plot - absolute values</span></span>
<span><span class="co"># comparison_plot(calc_dat = calc_dat_1,</span></span>
<span><span class="co">#                 theoretical = FALSE,</span></span>
<span><span class="co">#                 relative = FALSE,</span></span>
<span><span class="co">#                 state_first = "CD160",</span></span>
<span><span class="co">#                 state_second = "CD160_HVEM")</span></span></code></pre></div>
<p>The plot below shows peptides for which levels of deuteration were
significantly lower upon binding with other protein (red) and peptides
for which levels of exchange were significantly higher upon binding with
other protein (blue). The plot also shows peptides in which no
significant changes in deuteration between states are visible (grey).
The biggest changes on the theoretical plot can be observed in 3
peptides (15-24, 30-35, 75-90).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># theoretical Woods plot - relative values</span></span>
<span><span class="co"># woods_plot(calc_dat = calc_dat_1,</span></span>
<span><span class="co">#            theoretical = TRUE, </span></span>
<span><span class="co">#            relative = TRUE) +</span></span>
<span><span class="co">#   coord_cartesian(ylim = c(-.2, .2))</span></span></code></pre></div>
<p>The biggest changes on the experimental plot can be observed in 3
peptides (15-24, 110-115).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># experimental Woods plot - relative values</span></span>
<span><span class="co"># woods_plot(calc_dat = calc_dat_1,</span></span>
<span><span class="co">#            theoretical = FALSE,</span></span>
<span><span class="co">#            relative = TRUE) +</span></span>
<span><span class="co">#   coord_cartesian(ylim = c(-.2, .2))</span></span></code></pre></div>
<p>Plot below shows results in absolute values.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># theoretical Woods plot - absolute values</span></span>
<span><span class="co"># woods_plot(calc_dat = calc_dat_1,</span></span>
<span><span class="co">#            theoretical = TRUE, </span></span>
<span><span class="co">#            relative = FALSE) +</span></span>
<span><span class="co">#   labs(title = "Theoretical fraction exchanged between states in 1 min time") </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # experimental Woods plot - absolute values</span></span>
<span><span class="co"># woods_plot(calc_dat = calc_dat_1,</span></span>
<span><span class="co">#            theoretical = FALSE,</span></span>
<span><span class="co">#            relative = FALSE) +</span></span>
<span><span class="co">#   labs(title = "Fraction exchanged between states in 1 min time") </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # quality control - relative values </span></span>
<span><span class="co"># (result &lt;- quality_control(dat = dat_1,</span></span>
<span><span class="co">#                 state_first = "CD160",</span></span>
<span><span class="co">#                 state_second = "CD160_HVEM", </span></span>
<span><span class="co">#                 chosen_time = 1, </span></span>
<span><span class="co">#                 in_time = 0.001))</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # example quality control visualisation</span></span>
<span><span class="co"># library(ggplot2)</span></span>
<span><span class="co"># ggplot(result) + </span></span>
<span><span class="co">#   geom_line(aes(x = out_time, y = avg_err_state_first, color = "Average error (first state)")) +</span></span>
<span><span class="co">#   geom_line(aes(x = out_time, y = avg_err_state_second, color = "Average error (second state)")) +</span></span>
<span><span class="co">#   scale_x_log10() +</span></span>
<span><span class="co">#   ylim(0, 0.05) + </span></span>
<span><span class="co">#   labs(x = "log(time) [min]", y = "Average uncertainty", title = "Uncertainty change in out time") +</span></span>
<span><span class="co">#   theme(legend.title = element_blank(),</span></span>
<span><span class="co">#         legend.position = "bottom")</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-workflow-2">Example workflow 2<a class="anchor" aria-label="anchor" href="#example-workflow-2"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://hadexversum.github.io/HaDeX2/">HaDeX</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># # file import</span></span>
<span><span class="co"># dat_2 &lt;-  read_hdx(system.file(package = "HaDeX",                                "HaDeX/data/KD_190304_Nucb2_EDTA_CaCl2_test02_clusterdata.csv"))</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # protein sequence reconstruction</span></span>
<span><span class="co"># reconstruct_sequence(dat_2)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # calculate data</span></span>
<span><span class="co"># calc_dat_2 &lt;- prepare_dataset(dat = dat_2,</span></span>
<span><span class="co">#                               in_state_first = "gg_Nucb2_EDTA_0.001",</span></span>
<span><span class="co">#                               chosen_state_first = "gg_Nucb2_EDTA_25",</span></span>
<span><span class="co">#                               out_state_first = "gg_Nucb2_EDTA_1440",</span></span>
<span><span class="co">#                               in_state_second = "gg_Nucb2_CaCl2_0.001",</span></span>
<span><span class="co">#                               chosen_state_second = "gg_Nucb2_CaCl2_25",</span></span>
<span><span class="co">#                               out_state_second = "gg_Nucb2_CaCl2_1440")                             </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # theoretical comparison plot - relative values</span></span>
<span><span class="co"># comparison_plot(calc_dat = calc_dat_2,</span></span>
<span><span class="co">#                 theoretical = TRUE,</span></span>
<span><span class="co">#                 relative = TRUE,</span></span>
<span><span class="co">#                 state_first = "Nucb2 Factor 1",</span></span>
<span><span class="co">#                 state_second = "Nucb2 Factor 2") +</span></span>
<span><span class="co">#   labs(title = "Theoretical fraction exchanged in \nstate comparison in 25 min time")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # experimental comparison plot - relative values </span></span>
<span><span class="co"># comparison_plot(calc_dat = calc_dat_2,</span></span>
<span><span class="co">#                 theoretical = FALSE,</span></span>
<span><span class="co">#                 relative = TRUE,</span></span>
<span><span class="co">#                 state_first = "Nucb2 Factor 1",</span></span>
<span><span class="co">#                 state_second = "Nucb2 Factor 2") +</span></span>
<span><span class="co">#   labs(title = "Fraction exchanged in \nstate comparison in 25 min time")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # theoretical comparison plot - absolute values</span></span>
<span><span class="co"># comparison_plot(calc_dat = calc_dat_2,</span></span>
<span><span class="co">#                 theoretical = TRUE,</span></span>
<span><span class="co">#                 relative = FALSE,</span></span>
<span><span class="co">#                 state_first = "Nucb2 Factor 1",</span></span>
<span><span class="co">#                 state_second = "Nucb2 Factor 2") +</span></span>
<span><span class="co">#   labs(title = "Theoretical fraction exchanged in \nstate comparison in 25 min time")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # experimental comparison plot - absolute values </span></span>
<span><span class="co"># comparison_plot(calc_dat = calc_dat_2,</span></span>
<span><span class="co">#                 theoretical = FALSE,</span></span>
<span><span class="co">#                 relative = FALSE,</span></span>
<span><span class="co">#                 state_first = "Nucb2 Factor 1",</span></span>
<span><span class="co">#                 state_second = "Nucb2 Factor 2") +</span></span>
<span><span class="co">#   labs(title = "Fraction exchanged in \nstate comparison in 25 min time")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # theoretical Woods plot - relative values</span></span>
<span><span class="co"># woods_plot(calc_dat = calc_dat_2,</span></span>
<span><span class="co">#            theoretical = TRUE,</span></span>
<span><span class="co">#            relative = TRUE) + </span></span>
<span><span class="co">#   labs(title = "Theoretical fraction exchanged between states in 25 min time") +</span></span>
<span><span class="co">#   coord_cartesian(ylim = c(-.5, .7))</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # experimental Woods plot - relative values</span></span>
<span><span class="co"># woods_plot(calc_dat = calc_dat_2,</span></span>
<span><span class="co">#            theoretical = FALSE,</span></span>
<span><span class="co">#            relative = TRUE) +</span></span>
<span><span class="co">#   labs(title = "Fraction exchanged between states in 25 min time") +</span></span>
<span><span class="co">#   coord_cartesian(ylim = c(-.5, .7))</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # theoretical Woods plot - absolute values</span></span>
<span><span class="co"># woods_plot(calc_dat = calc_dat_2,</span></span>
<span><span class="co">#            theoretical = TRUE,</span></span>
<span><span class="co">#            relative = FALSE) + </span></span>
<span><span class="co">#   labs(title = "Theoretical fraction exchanged between states in 25 min time") </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # experimental Woods plot - absolute values</span></span>
<span><span class="co"># woods_plot(calc_dat = calc_dat_2,</span></span>
<span><span class="co">#            theoretical = FALSE,</span></span>
<span><span class="co">#            relative = FALSE) +</span></span>
<span><span class="co">#   labs(title = "Fraction exchanged between states in 25 min time") </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # quality control</span></span>
<span><span class="co"># (result &lt;- quality_control(dat = dat_2,</span></span>
<span><span class="co">#                            state_first = "gg_Nucb2_EDTA",</span></span>
<span><span class="co">#                            state_second = "gg_Nucb2_CaCl2", </span></span>
<span><span class="co">#                            chosen_time = 25, </span></span>
<span><span class="co">#                            in_time = 0.001))</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # example quality control visualisation - relative values</span></span>
<span><span class="co"># library(ggplot2)</span></span>
<span><span class="co"># ggplot(result[result["out_time"]&gt;=1,]) + </span></span>
<span><span class="co">#   geom_line(aes(x = out_time, y = avg_err_state_first, color = "Average error (first state)")) +</span></span>
<span><span class="co">#   geom_line(aes(x = out_time, y = avg_err_state_second, color = "Average error (second state)")) +</span></span>
<span><span class="co">#   scale_x_log10() +</span></span>
<span><span class="co">#   labs(x = "log(time) [min]", y = "Average uncertainty", title = "Uncertainty change") + </span></span>
<span><span class="co">#   theme(legend.position = "bottom",</span></span>
<span><span class="co">#         legend.title = element_blank())</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p><!-- ```{r warning=FALSE} --> <!-- # calculate data -->
<!-- calc_dat_1 <- prepare_dataset(dat = dat_1, -->
<!--                               in_state_first = "CD160_0.001", -->
<!--                               chosen_state_first = "CD160_1", -->
<!--                               out_state_first = "CD160_1440", -->
<!--                               in_state_second = "CD160_HVEM_0.001", -->
<!--                               chosen_state_second = "CD160_HVEM_1", -->
<!--                               out_state_second = "CD160_HVEM_1440")                              --></p>
<p><!-- # theoretical comparison plot - relative values -->
<!-- comparison_plot(calc_dat = calc_dat_1, -->
<!--                 theoretical = TRUE, -->
<!--                 relative = TRUE,  -->
<!--                 state_first = "CD160", -->
<!--                 state_second = "CD160_HVEM") --></p>
<p><!-- ``` --></p>
<p>On the experimental plot, there are visible regions, which exchange
quickly (N terminal part, regions between 30-70 amino acid) and regions,
which exchange slowly (peptides 15-24, 30-35, 95-110 amino acid).</p>
<p><!-- ```{r warning=FALSE} -->
<!-- # experimental comparison plot - relative values -->
<!-- comparison_plot(calc_dat = calc_dat_1, -->
<!--                 theoretical = FALSE, -->
<!--                 relative = TRUE, -->
<!--                 state_first = "CD160", -->
<!--                 state_second = "CD160_HVEM") --></p>
<p><!-- ``` --></p>
<p>Plots below are equivalent to plots above but in absolute values -
for users that prefer those.</p>
<p><!-- ```{r warning=FALSE} -->
<!-- # theoretical comparison plot - absolute values -->
<!-- comparison_plot(calc_dat = calc_dat_1, -->
<!--                 theoretical = TRUE, -->
<!--                 relative = FALSE,  -->
<!--                 state_first = "CD160", -->
<!--                 state_second = "CD160_HVEM") --></p>
<p><!-- # experimental comparison plot - absolute values -->
<!-- comparison_plot(calc_dat = calc_dat_1, -->
<!--                 theoretical = FALSE, -->
<!--                 relative = FALSE, -->
<!--                 state_first = "CD160", -->
<!--                 state_second = "CD160_HVEM") --> <!-- ``` --></p>
<p>The plot below shows peptides for which levels of deuteration were
significantly lower upon binding with other protein (red) and peptides
for which levels of exchange were significantly higher upon binding with
other protein (blue). The plot also shows peptides in which no
significant changes in deuteration between states are visible (grey).
The biggest changes on the theoretical plot can be observed in 3
peptides (15-24, 30-35, 75-90).</p>
<p><!-- ```{r warning=FALSE} --></p>
<p><!-- # theoretical Woods plot - relative values -->
<!-- woods_plot(calc_dat = calc_dat_1, -->
<!--            theoretical = TRUE,  -->
<!--            relative = TRUE) + -->
<!--   coord_cartesian(ylim = c(-.2, .2)) --></p>
<p><!-- ``` --></p>
<p>The biggest changes on the experimental plot can be observed in 3
peptides (15-24, 110-115).</p>
<!-- # ```{r warning=FALSE} -->
<!-- # # experimental Woods plot - relative values -->
<!-- # woods_plot(calc_dat = calc_dat_1, -->
<!-- #            theoretical = FALSE, -->
<!-- #            relative = TRUE) + -->
<!-- #   coord_cartesian(ylim = c(-.2, .2)) -->
<!-- #  -->
<!-- # ``` -->
<p>Plot below shows results in absolute values.</p>
<!-- # ```{r warning=FALSE} -->
<!-- # # theoretical Woods plot - absolute values -->
<!-- # woods_plot(calc_dat = calc_dat_1, -->
<!-- #            theoretical = TRUE,  -->
<!-- #            relative = FALSE) + -->
<!-- #   labs(title = "Theoretical fraction exchanged between states in 1 min time")  -->
<!-- #  -->
<!-- # # experimental Woods plot - absolute values -->
<!-- # woods_plot(calc_dat = calc_dat_1, -->
<!-- #            theoretical = FALSE, -->
<!-- #            relative = FALSE) + -->
<!-- #   labs(title = "Fraction exchanged between states in 1 min time")  -->
<!-- #  -->
<!-- # # quality control - relative values  -->
<!-- # (result <- quality_control(dat = dat_1, -->
<!-- #                 state_first = "CD160", -->
<!-- #                 state_second = "CD160_HVEM",  -->
<!-- #                 chosen_time = 1,  -->
<!-- #                 in_time = 0.001)) -->
<!-- #  -->
<!-- # # example quality control visualisation -->
<!-- # library(ggplot2) -->
<!-- # ggplot(result) +  -->
<!-- #   geom_line(aes(x = out_time, y = avg_err_state_first, color = "Average error (first state)")) + -->
<!-- #   geom_line(aes(x = out_time, y = avg_err_state_second, color = "Average error (second state)")) + -->
<!-- #   scale_x_log10() + -->
<!-- #   ylim(0, 0.05) +  -->
<!-- #   labs(x = "log(time) [min]", y = "Average uncertainty", title = "Uncertainty change in out time") + -->
<!-- #   theme(legend.title = element_blank(), -->
<!-- #         legend.position = "bottom") -->
<!-- #  -->
<!-- # ``` -->
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-BouyssieHDXViewerinteractive3D2019" class="csl-entry">
Bouyssié, David, Jean Lesne, Marie Locard-Paulet, Renaud Albigot, Odile
Burlet-Schiltz, and Julien Marcoux. 2019.
<span>“<span>HDX</span>-<span>Viewer</span>: Interactive <span>3D</span>
Visualization of Hydrogen-Deuterium Exchange Data.”</span>
<em>Bioinformatics (Oxford, England)</em>, July. <a href="https://doi.org/10.1093/bioinformatics/btz550" class="external-link">https://doi.org/10.1093/bioinformatics/btz550</a>.
</div>
<div id="ref-HoudeUtilityHydrogenDeuterium2011" class="csl-entry">
Houde, Damian, Steven A. Berkowitz, and John R. Engen. 2011. <span>“The
<span>Utility</span> of <span>Hydrogen</span>/<span>Deuterium Exchange
Mass Spectrometry</span> in <span>Biopharmaceutical Comparability
Studies</span>.”</span> <em>Journal of Pharmaceutical Sciences</em> 100
(6): 2071–86. <a href="https://doi.org/10.1002/jps.22432" class="external-link">https://doi.org/10.1002/jps.22432</a>.
</div>
<div id="ref-jcgm:2008:EMDG" class="csl-entry">
Joint Committee for Guides in Metrology. 2008. <span>“<span>JCGM</span>
100: <span>Evaluation</span> of <span>Measurement Data</span> -
<span>Guide</span> to the <span>Expression</span> of
<span>Uncertainty</span> in <span>Measurement</span>.”</span>
<span>JCGM</span>.
</div>
<div id="ref-MassonRecommendationsperforminginterpreting2019" class="csl-entry">
Masson, Glenn R., John E. Burke, Natalie G. Ahn, Ganesh S. Anand,
Christoph Borchers, Sébastien Brier, George M. Bou-Assaf, et al. 2019.
<span>“Recommendations for Performing, Interpreting and Reporting
Hydrogen Deuterium Exchange Mass Spectrometry
(<span>HDX</span>-<span>MS</span>) Experiments.”</span> <em>Nature
Methods</em> 16 (7): 595–602. <a href="https://doi.org/10.1038/s41592-019-0459-y" class="external-link">https://doi.org/10.1038/s41592-019-0459-y</a>.
</div>
<div id="ref-PascalHDXWorkbenchSoftware2012" class="csl-entry">
Pascal, Bruce D., Scooter Willis, Janelle L. Lauer, Rachelle R.
Landgraf, Graham M. West, David Marciano, Scott Novick, Devrishi
Goswami, Michael J. Chalmers, and Patrick R. Griffin. 2012.
<span>“<span>HDX Workbench</span>: <span>Software</span> for the
<span>Analysis</span> of <span>H</span>/<span>D Exchange MS
Data</span>.”</span> <em>Journal of The American Society for Mass
Spectrometry</em> 23 (9): 1512–21. <a href="https://doi.org/10.1007/s13361-012-0419-6" class="external-link">https://doi.org/10.1007/s13361-012-0419-6</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Weronika Puchala, Michal Burdukiewicz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
