<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Basics of HDX-MS â€¢ HaDeX</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Basics of HDX-MS">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HaDeX</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/HaDeX.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/datafiles.html">Data processing</a></li>
    <li><a class="dropdown-item" href="../articles/glossary.html">Glossary</a></li>
    <li><a class="dropdown-item" href="../articles/hdxms.html">Basics of HDX-MS</a></li>
    <li><a class="dropdown-item" href="../articles/mass_comparison.html">Mass calculation</a></li>
    <li><a class="dropdown-item" href="../articles/transformation.html">Data aggregation</a></li>
    <li><a class="dropdown-item" href="../articles/usage.html">Code usage</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Data visualization</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Basics of HDX-MS</h1>
            
      

      <div class="d-none name"><code>hdxms.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction-to-hdx-ms">Introduction to HDX-MS<a class="anchor" aria-label="anchor" href="#introduction-to-hdx-ms"></a>
</h2>
<p>Hydrogen-deuterium exchange mass spectrometry (HDX-MS) is a way to
perform an analysis of a protein. The principal is to observe the
behavior of the protein in the native state in the aqueous solution
where the hydrogen particles are replaced by the deuterium particles.
Isotopes (variants of the same chemical element but with the different
number of neutrons) exchanges with themselves very well. Under
controlled conditions, there is a spontaneous process of exchange. This
kind of experiment can gather very effective data on how the exchange
goes in the main chain of the protein - the exchange on the ends of the
sequence goes too fast to be correctly measured. This kind of data not
only serves the purpose of gaining the basic information about protein
structure but also getting some insight into the protein structure of
the tertiary and quaternary order.</p>
<p>What is special about HDX-MS experiments is that they produce unique
data that cannot be obtained in any other way. For example, nearly
one-third of the proteins are considered as intrinsically disordered
protein - lacking a fixed or ordered three-dimensional structure. For
those, the crystallography experiments cannot be performed - but HDX-MS
can. It is also a suitable solution for analysis of very long proteins -
compared to short protein sequences that can be processed using the NMR
method. Moreover, HDX-MS can give some insight if the function of the
protein is preserved when joined with other substances - that is
particularly important when constructing and analyzing new or already
existing drugs.</p>
<p>The disadvantage of the HDX-MS method in comparison with
crystallography or NMR is gaining data for whole peptide (considered as
low resolution) instead of a single amino residue (high resolution).
Elimination of this limitation could make HDX-MS method more efficient
and competitive with others.</p>
<p>Data from HDX-MS experiments provide a unique insight into the
dynamics of protein structure. Such information is a very valuable
addition to existing or upcoming models of proteins and protein
complexes. They provide data on the mechanics of molecular biological
processes, where their dynamics are crucial for understanding their
course. However, we believe that HDX-MS data can be analyzed more
efficiently and informatively by switching from peptide (low-resolution)
to amino acid pairs (high-resolution).</p>
</div>
<div class="section level2">
<h2 id="additional-resources">Additional resources<a class="anchor" aria-label="anchor" href="#additional-resources"></a>
</h2>
<p>To learn more about HDX-MS, we propose:</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Weronika Puchala, Michal Burdukiewicz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
