<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Data visualization • HaDeX2</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Data visualization">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HaDeX2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/data_input.html">Importing data</a></li>
    <li><a class="dropdown-item" href="../articles/datafiles.html">Calculations</a></li>
    <li><a class="dropdown-item" href="../articles/example.html">Example HDX-MS data analysis</a></li>
    <li><a class="dropdown-item" href="../articles/glossary.html">Glossary</a></li>
    <li><a class="dropdown-item" href="../articles/mass_comparison.html">Mass calculation</a></li>
    <li><a class="dropdown-item" href="../articles/quality_control.html">Quality control</a></li>
    <li><a class="dropdown-item" href="../articles/statistics.html">Statistics</a></li>
    <li><a class="dropdown-item" href="../articles/transformation.html">Data aggregation</a></li>
    <li><a class="dropdown-item" href="../articles/usage.html">Code usage</a></li>
    <li><a class="dropdown-item" href="../articles/version_comparison.html">Comparison between versions</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Data visualization</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="visualization_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="visualization_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="visualization_files/r3dmol-1.6.2/3Dmol-nojquery-min.js"></script><script src="visualization_files/r3dmol-binding-0.2.0/r3dmol.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Data visualization</h1>
            
      

      <div class="d-none name"><code>visualization.Rmd</code></div>
    </div>

    
    
<p>Data visualization is a crucial part of the experimental data
analysis. The forms of visualization should be adjusted to highlight the
essential result and tailored to satisfy personal needs.</p>
<p>In this article, we present methods of data visualization available
in the HaDeX2 package, and consecutively in the GUI.</p>
<p>The analyzed protein is the eEF1Bα subunit of the human
guanine-nucleotide exchange factor (GEF) complex (eEF1B), measured in <a href="https://mslab-ibb.pl/" class="external-link">Mass Spectrometry Lab</a> in <a href="https://ibb.edu.pl/en/" class="external-link">Institute of Biochemistry and Biophysics
Polish Academy of Sciences</a> and published by <a href="https://doi.org/10.1093/nar/gkac685" class="external-link">Bondarchuk et al</a>. In the
one-state classification, we will focus on pure alpha state -
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>.
The comparative analysis is conducted between
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
and
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
in presence of
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>γ</mi><annotation encoding="application/x-tex">\gamma</annotation></semantics></math>.</p>
<div class="section level2">
<h2 id="comparison-plot">Comparison plot<a class="anchor" aria-label="anchor" href="#comparison-plot"></a>
</h2>
<p>The comparison plot presents deuterium uptake of the peptides in a
given time point, with information on the length of the peptide and
their position in the protein sequence. It allows comparison of the
results of different biological states.</p>
<p><strong>Example</strong> In the comparison plot below, we see the
fractional deuterium uptake for all three possible states:
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>,
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
in presence of
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>γ</mi><annotation encoding="application/x-tex">\gamma</annotation></semantics></math>
and in presence of
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>β</mi><annotation encoding="application/x-tex">\beta</annotation></semantics></math>,
from protein eEF1B. The values are calculated for the time point 1 min.
The length of the segments represents the length of the peptide and the
position in the protein sequence. The error bars indicate the
uncertainty of the measurement.</p>
<pre><code><span><span class="co">## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">HaDeX2</span> package.</span></span>
<span><span class="co">##   Please report the issue to the authors.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<p><img src="visualization_files/figure-html/unnamed-chunk-2-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>peptide length and position</li>
<li>comparison of multiple biological states</li>
<li>region with difference is easy to spot on</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
<li>plots with different time points can be plotted next to each
other</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="woods-plot">Woods plot<a class="anchor" aria-label="anchor" href="#woods-plot"></a>
</h2>
<p>Woods plot presents the deuterium uptake difference between two
biological states for the peptides. The results are presented with
respect to the length of the peptide and its position in the protein
sequence for a given time point of the measurement. The statistical test
(Houde et al., the confidence level is calculated based on the
measurements in chosen time point of the measurement) is applied to
determine the confidence limits values at the chosen level.</p>
<p><strong>Example</strong> On the Woods plot below, we see fractional
deuterium uptake difference between two biological states
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
and
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
in presence of
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>γ</mi><annotation encoding="application/x-tex">\gamma</annotation></semantics></math>
for protein eEF1B The confidence limits indicate which differences are
statistically significant at levels 98%.</p>
<p><img src="visualization_files/figure-html/unnamed-chunk-3-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>differential analysis between two states</li>
<li>peptide length and position</li>
<li>region with difference is easy to spot on</li>
<li>uncertainty for each peptide</li>
<li>hybrid statistical testing</li>
<li>statistical level</li>
</ul>
<p><em>Plot variables:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
<li>plots with different time points can be plotted next to each
other</li>
<li>peptides classified as statistically insignificant can be
hidden</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="butterfly-plot">Butterfly plot<a class="anchor" aria-label="anchor" href="#butterfly-plot"></a>
</h2>
<p>Butterfly plot presents the deuterium uptake for all peptides in a
given state at different time points at once. Each time point of
measurement is indicated by a different color. Peptides are identified
by their ID (peptides are numbered arranged by the start position).</p>
<p><strong>Example</strong> Below, on the butterfly plot, we see how the
deuterium uptake changes in time for state
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
for protein eEF1B. We see the different exchange speed - for some
peptides, the change is stable in time, and for some peptides, there is
no visible change in time.</p>
<p><img src="visualization_files/figure-html/unnamed-chunk-4-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>values for time course</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>only one state</li>
<li>lost information of length and position of the peptide</li>
<li>may be difficult to read</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
<li>different methods of showing uncertainty: bars or ribbons</li>
<li>selected time points shown</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="butterfly-differential-plot">Butterfly differential plot<a class="anchor" aria-label="anchor" href="#butterfly-differential-plot"></a>
</h2>
<p>Butterfly differential plot shows the deuterium uptake difference
between two biological states in the form of a butterfly plot. It shows
the results for a peptide ID (peptides are numbered arranged by the
start position). The results are shown for different time points at once
(time points of measurement are indicated by the color).</p>
<p><strong>Example</strong> Below, we see how the fractional deuterium
uptake difference between states
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
and
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
in presence of
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>γ</mi><annotation encoding="application/x-tex">\gamma</annotation></semantics></math>
changes over time. We see that for some peptides, the difference is
smaller with time - perhaps because of the back exchange.</p>
<p>The measurements for 1440 min are hidden, as they are close to 0, as
expected.</p>
<p><img src="visualization_files/figure-html/unnamed-chunk-5-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>differential analysis between two states</li>
<li>hybrid statistical testing</li>
<li>statistical significance level</li>
<li>values for time course</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>lost information of length and position of the peptide</li>
<li>may be difficult to read</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
<li>different methods of showing uncertainty: bars or ribbons</li>
<li>selected time points shown</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="chiclet-plot">Chiclet plot<a class="anchor" aria-label="anchor" href="#chiclet-plot"></a>
</h2>
<p>Chiclet plot shows the fractional deuterium uptake in the form of a
heatmap for the peptides in a given biological state. One tile indicates
the peptide (identified by its ID - number arranged by the start
position) in a time point of measurement. The color of the tile
indicates the fractional deuterium uptake (according to the legend below
the plot).</p>
<p><strong>Example</strong> In the chiclet plot below, we can see the
deuterium uptake values for peptides (indicated by their ID) in state
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
during the time course of the experiment. The cross symbols indicate the
uncertainty of the measurement (the bigger the cross sign, the bigger
the uncertainty).</p>
<pre><code><span><span class="co">## Warning: The `size` argument of `element_rect()` is deprecated as of ggplot2 3.4.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">HaDeX2</span> package.</span></span>
<span><span class="co">##   Please report the issue to the authors.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<p><img src="visualization_files/figure-html/unnamed-chunk-6-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>values for time course</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>only one state</li>
<li>lost information of length and position of the peptide</li>
<li>may be difficult to read</li>
<li>small changes in values may be difficult to spot</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
<li>uncertainty can be hidden to improve the readability of the
plot</li>
<li>peptides classified as statistically significant can be hidden</li>
<li>selected time points shown</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="chiclet-differential-plot">Chiclet differential plot<a class="anchor" aria-label="anchor" href="#chiclet-differential-plot"></a>
</h2>
<p>Chiclet differential plot shows the deuterium uptake difference
between two biological states in the form of a heatmap. One tile
indicates the peptide (identified by its ID - number arranged by the
start position) in a time point of measurement. The color of the tile
indicates the deuterium uptake difference (according to the legend below
the plot).</p>
<p><strong>Example</strong> On the chiclet differential plot below, we
see the fractional deuterium uptake difference between states
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
and
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
in presence of
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>γ</mi><annotation encoding="application/x-tex">\gamma</annotation></semantics></math>
for protein eEF1B. We see that some peptides are protected (red), and
some are deprotected (blue). The cross symbols indicate the uncertainty
of the measurement (the bigger the cross sign, the bigger the
uncertainty).</p>
<p><img src="visualization_files/figure-html/unnamed-chunk-7-1.png" class="r-plt" width="672"><em>Pros:</em></p>
<ul>
<li>differential analysis between two states</li>
<li>values for time course</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>lost information of length and position of the peptide</li>
<li>may be difficult to read</li>
<li>small changes in values may be difficult to spot</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
<li>uncertainty can be hidden to improve the readability of the
plot</li>
<li>peptides classified as statistically significant can be hidden</li>
<li>selected time points shown</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="volcano-plot">Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h2>
<p>The volcano plot shows the deuterium uptake difference for two
biological states for peptide and its p-value for double testing on
statistical significance (Weis et al.). On the x-axis, there is a
deuterium uptake difference with its uncertainty (combined and
propagated). On the y-axis, there is a P-value calculated for each
peptide in a specific time point of a measurement as a un-paired t-test
on given significance level (on mass measurement from the replicates to
indicate if the measured mean is significantly different between two
states, as the deuterium uptake difference between states can be
rewritten as</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Δ</mi><mi>D</mi><mo>=</mo><msub><mi>D</mi><mi>A</mi></msub><mo>−</mo><msub><mi>D</mi><mi>B</mi></msub><mo>=</mo><msub><mi>m</mi><mrow><mi>t</mi><mo>,</mo><mi>A</mi></mrow></msub><mo>−</mo><msub><mi>m</mi><mn>0</mn></msub><mo>−</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>m</mi><mrow><mi>t</mi><mo>,</mo><mi>B</mi></mrow></msub><mo>−</mo><msub><mi>m</mi><mn>0</mn></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><msub><mi>m</mi><mrow><mi>t</mi><mo>,</mo><mi>A</mi></mrow></msub><mo>−</mo><msub><mi>m</mi><mrow><mi>t</mi><mo>,</mo><mi>B</mi></mrow></msub></mrow><annotation encoding="application/x-tex">\Delta D = D_{A} - D_{B} = m_{t, A} - m_{0} - (m_{t, B} - m_{0}) = m_{t, A} - m_{t, B} </annotation></semantics></math></p>
<p>for states A and B. The values of deuterium uptake difference from
all time points are shown on the plot.</p>
<p>The dotted red lines indicate confidence limits for the values. The
horizontal line indicates the confidence limit based on chosen
confidence level to give a threshold on a P-value. The vertical lines
indicate the confidence limit from Houde test for all time points and
indicate a threshold on deuterium uptake difference. The statistically
significant points are in the top left and right corners of the
plot.</p>
<p><strong>Example</strong> On the volcano plot below, we see the
results for deuterium uptake difference between states
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
and
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
in presence of
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>γ</mi><annotation encoding="application/x-tex">\gamma</annotation></semantics></math>
for protein eEF1B in all time points. The points in the left and right
upper corner are statistically significant using the hybrid testing.</p>
<p><img src="visualization_files/figure-html/unnamed-chunk-8-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>hybrid statistic test</li>
<li>time course information</li>
<li>uncertainty</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>peptide length and position lost</li>
<li>no information of region</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>selected time points</li>
<li>absolute/fractional values s</li>
<li>hidden insignificant values</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="uptake-curve">Uptake curve<a class="anchor" aria-label="anchor" href="#uptake-curve"></a>
</h2>
<p>Uptake curves show the changes in exchange in time for a specific
peptide for its state.</p>
<p><strong>Example</strong> On the uptake curve below, we see how the
exchange goes for peptide GFGDLKSPAGL in all three states for protein
eEF1Ba.</p>
<p><img src="visualization_files/figure-html/unnamed-chunk-9-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>time course</li>
<li>state comparison</li>
<li>change of uptake tendency visible</li>
<li>focus on one peptide</li>
<li>uncertainty</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
<li>selected states shown</li>
<li>different methods of showing uncertainty: bars or ribbons</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="uncertainty-plot">Uncertainty plot<a class="anchor" aria-label="anchor" href="#uncertainty-plot"></a>
</h2>
<p>Uncertainty plot is new visualization method, showing the uncertainty
of measurement of deuterium uptake for peptides to spot the regions
where the uncertainty is higher. This plot may be used as a quality
control of the experiment, as discussed in the
<code>vignette(quality_control)</code>. The presented uncertainty is in
Daltons, making the threshold 1 Da as proposed limit of acceptance.</p>
<p><strong>Example</strong> The plot below presents uncertainty values
for multiple time points for state
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>.
We see that the uncertainty is relatively low for all the measurement,
and none of the value is suspicious.</p>
<pre><code><span><span class="co">## Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_segment()`).</span></span></code></pre>
<p><img src="visualization_files/figure-html/unnamed-chunk-10-1.png" class="r-plt" width="672"><em>Pros:</em></p>
<ul>
<li>experimental quality control</li>
<li>regions with high measurement uncertainty are easy to spot</li>
<li>peptide length and position</li>
<li>region with difference is easy to spot on</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>plot for only one time point can be plotted</li>
<li>selected time points shown</li>
<li>threshold can be hidden</li>
<li>first amino omitted</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="manhattan-plot">Manhattan plot<a class="anchor" aria-label="anchor" href="#manhattan-plot"></a>
</h2>
<p>Manhattan plot is a novel plot, presenting the P-value of statistical
significance between two states.</p>
<p><strong>Example</strong> In this example, we present the P values of
difference between two biological states:
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
and
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
in presence of
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>γ</mi><annotation encoding="application/x-tex">\gamma</annotation></semantics></math>.
We can see the regions where the difference is statistically significant
- above the significance level (detailed as option in
create_p_diff_uptake_dataset, or by default 0.98).</p>
<pre><code><span><span class="co">## Warning: Removed 77 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_segment()`).</span></span></code></pre>
<p><img src="visualization_files/figure-html/unnamed-chunk-11-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>peptide length and position</li>
<li>shows significant regions</li>
<li>statistical interval shown</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>plot for only one time point can be plotted</li>
<li>selected time points shown</li>
<li>threshold can be hidden</li>
<li>first amino omitted</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="high-resolution-plot">High-resolution plot<a class="anchor" aria-label="anchor" href="#high-resolution-plot"></a>
</h2>
<p>The biggest limitation of previous methods of deuterium uptake
visualization is that the results are on the peptide level. However, we
offer a method of aggregation of data on peptide level into
high-resolution level, using the weighted method of aggregation (Keppler
and Weis (doi: 10.1007/s13361-014-1033-6)). Then, the results are
presented on the heatmap.</p>
<p><strong>Example</strong></p>
<p><img src="visualization_files/figure-html/unnamed-chunk-12-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>high-resolution</li>
<li>time course</li>
<li>uncertainty of aggregated values</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>values are an approximation</li>
<li>linear presentation lacking conformational information</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
<li>plot divided into panels</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="differential-high-resolution-plot">Differential High-resolution plot<a class="anchor" aria-label="anchor" href="#differential-high-resolution-plot"></a>
</h2>
<p>As most of our methods of visualization, also the high-resolution
plot has its differential version. The plot presents aggregated uptake
difference values using weighted approach.</p>
<p><strong>Example</strong> On the structure, we see the fractional
deuterium uptake difference after 25 minutes between states
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
and
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
in presence of
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>γ</mi><annotation encoding="application/x-tex">\gamma</annotation></semantics></math>
for protein eEF1B. We see that some peptides are protected (red), and
some are deprotected (blue).</p>
<p><img src="visualization_files/figure-html/unnamed-chunk-13-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>high-resolution</li>
<li>comparative analysis</li>
<li>time course</li>
<li>uncertainty of aggregated values</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>values are an approximation</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
<li>plot divided into panels</li>
<li>tooltips available in GUI</li>
</ul>
<!-- # Differential High-resolution on 3D structure --><!-- Differential aggregated results can be presented on 3d structure, if available. This plot has a very limited options for personalization, but is extremly vital for comparison of experimental uptake data with spatial information about conformation of the protein. --><!-- **Example**  On the structure, we see the fractional deuterium uptake difference after 25 minutes between states eEF1B$\alpha$ and eEF1B$\alpha$ in presence of eEF1B$\gamma$ for protein eEF1B. We see that some peptides are protected (red), and some are deprotected (blue). --><!-- ```{r} --><!-- pdb_file_path <- system.file(package = "HaDeX", "HaDeX/data/Model_eEF1Balpha.pdb") --><!-- plot_aggregated_uptake_structure(aggregated_diff_dat,  --><!--                                  differential = TRUE, --><!--                                  time_t = 25, --><!--                                  pdb_file_path = pdb_file_path) --><!-- ``` --><!-- *Pros:* --><!--  * aggregated values --><!--  * spatial information --><!--  * spinning, rotating and zooming options --><!-- *Cons:* --><!--  * only one time point at a time --><!--  * no personalization --><!--  * no information on numerical uptake values --><!--  * no legend --><!-- *Plot variations:* --><!--  * fractional/absolute values --><!--  * fractional values calculated with respect of theoretical/experimental maximal uptake -->
</div>
<div class="section level2">
<h2 id="high-resolution-on-3d-structure">High-resolution on 3D structure<a class="anchor" aria-label="anchor" href="#high-resolution-on-3d-structure"></a>
</h2>
<p>High-resolution values not only can be presented linearly on
high-resolution plot (as above), but also on 3D structure, if available.
This way, the calculated values are connected with spatial aspect. This
option is available both for single state uptake and differential
uptake.</p>
<p><strong>Example</strong> Structure below is mapped with deuterium
uptake values after 1 minute for
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
state. Values are aggregateed using weighted approach, color signifies
no exchange (white) to high exchange (red).</p>
<div class="r3dmol html-widget html-fill-item" id="blank_structure" style="width:100%;height:500px;"></div>
<script type="application/json" data-for="blank_structure">{"x":{"id":"blank_structure","configs":{"backgroundColor":"#FFFFFF","nomouse":false,"backgroundColor.1":"white","cartoonQuality":10,"lowerZoomLimit":50,"upperZoomLimit":350,"antialias":true,"orthographic":false,"disableFog":false},"api":[{"method":"addModel","data":[""],"format":"pdb","keepH":false,"options":{"keepH":false}},{"method":"setStyle","style":{"cartoon":{"color":"white","style":"rectangle","ribbon":false,"arrows":true,"tubes":false,"thickness":0.4,"opacity":1}},"sel":{}},{"method":"zoomTo"},{"method":"setStyle","style":{"cartoon":{"style":"rectangle","ribbon":false,"arrows":true,"tubes":false,"thickness":0.4,"opacity":1,"colorfunc":"\n        function(atom) {\n          const color = [\"#FF9E81\",\"#FF6C4A\",\"#FF5838\",\"#FF5838\",\"#FF4628\",\"#FF4628\",\"#FF4628\",\"#FF4628\",\"#FF5031\",\"#FF5031\",\"#FF5938\",\"#FF8868\",\"#FF8868\",\"#FF8868\",\"#FF9475\",\"#FF9475\",\"#FF7D5D\",\"#FF7958\",\"#FF7553\",\"#FF7553\",\"#FF6241\",\"#FF4729\",\"#FF3E22\",\"#FF341A\",\"#FF341A\",\"#FF341A\",\"#FF3A1F\",\"#FF3A1F\",\"#FF3A1F\",\"#FF3C20\",\"#FF3C20\",\"#FF3C20\",\"#FF3E22\",\"#FFA387\",\"#FFA387\",\"#FFA387\",\"#FF7251\",\"#FF6A49\",\"#FF5635\",\"#FF5333\",\"#FF5333\",\"#FF5333\",\"#FF5333\",\"#FF5333\",\"#FF5333\",\"#FF5333\",\"#FF5333\",\"#FF5535\",\"#FF5636\",\"#FF5636\",\"#FF7B5A\",\"#FF9C7F\",\"#FF9C7F\",\"#FF9C7F\",\"#FF9C7F\",\"#FF9C7F\",\"#FF9C7F\",\"#FF230F\",\"#FF230F\",\"#FF230F\",\"#FF230F\",\"#FF230F\",\"#FF230F\",\"#FF1C0A\",\"#FF1C0A\",\"#FF1B0A\",\"#FF1B0A\",\"#FF1B0A\",\"#FF1B0A\",\"#FF1B0A\",\"#FF1B0A\",\"#FF1B0A\",\"#FF1B0A\",\"#FF1A09\",\"#FF1A09\",\"#FF1A09\",\"#FF1A09\",\"#FF1A09\",\"#FF1909\",\"#FF1909\",\"#FF1909\",\"#FF1909\",\"#FF1F0C\",\"#FF1F0C\",\"#FF2E16\",\"#FF2D16\",\"#FF2D16\",\"#FF2D16\",\"#FF2D16\",\"#FF2D16\",\"#FF2D16\",\"#FF2D16\",\"#FF2D16\",\"#FF2D16\",\"#FF2D16\",\"#FF2D16\",\"#FF2510\",\"#FF2510\",\"#FF2510\",\"#FF2510\",\"#FF2510\",\"#FF1F0C\",\"#FF2510\",\"#FF2510\",\"#FF2510\",\"#FF2510\",\"#FF2410\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF220E\",\"#FF210E\",\"#FF210D\",\"#FF210D\",\"#FF210E\",\"#FF210E\",\"#FF1F0D\",\"#FF1F0D\",\"#FF1F0D\",\"#FF1F0D\",\"#FF1F0D\",\"#FF1F0D\",\"#FF1F0D\",\"#FF1F0D\",\"#FF1F0D\",\"#FF1F0D\",\"#FF1F0D\",\"#FFAA90\",\"#FFA58A\",\"#FFA58A\",\"#FFA58A\",\"#FFA58A\",\"#FFA58A\",\"#FFA58A\",\"#FFB7A0\",\"#FFAD94\",\"#FFAD94\",\"#FFAD94\",\"#FFAD94\",\"#FFAD94\",\"#FFAD94\",\"#FFAD94\",\"#FF9A7D\",\"#FF9A7D\",\"#FF9A7D\",\"#FFCAB8\",\"#FFF0EB\",\"#FFF0EB\",\"#FFF0EB\",\"#FFF0EB\",\"#FFF0EB\",\"#FF9F82\",\"#FF997C\",\"#FF997C\",\"#FF8E6F\",\"#FF8E6F\",\"#FF8E6F\",\"#FF8E6F\",\"#FF8767\",\"#FF8767\",\"#FF8666\",\"#FF8A6B\",\"#FF8A6B\",\"#FF8A6B\",\"#FF8A6B\",\"#FF8A6B\",\"#FF8A6B\",\"#FF8A6B\",\"#FF9F82\",\"#FF9F82\",\"#FF9F82\",\"#FF9F82\",\"#FF9F82\",\"#FFA589\",\"#FFA98E\",\"#FFA98E\",\"#FFC8B6\",\"#FFD0C0\",\"#FFD0C0\",\"#FFD0C0\",\"#FFD0C0\",\"#FFC2AD\",\"#FF9C7F\",\"#FF9C7F\",\"#FF9C7F\",\"#FF9172\",\"#FF8868\",\"#FF8868\",\"#FF8464\",\"#FF8464\",\"#FF8464\",\"#FF8464\",\"#FF8464\",\"#FFB59D\",\"#FFE9E1\",\"#FFF0EA\",\"#FFF0EB\",\"#FFF0EB\",\"#FFF0EB\",\"#FFF1EC\",\"#FFEBE4\",\"#FFCFBF\",\"#FFC9B7\",\"#FFC8B5\",\"#FFC8B5\",\"#FFC8B5\",\"#FFC8B5\",\"#FFC8B5\",\"#FFB098\",\"#FF9B7E\",\"#FF8A6A\",\"#FF8A6A\",\"#FF8A6A\"];\n          return color[atom.resi];\n        }"}},"sel":{}}]},"evals":[],"jsHooks":[]}</script><p><em>Pros:</em></p>
<ul>
<li>aggregated values</li>
<li>spatial information</li>
<li>spinning, rotating and zooming options</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>only one time point at a time</li>
<li>no personalization</li>
<li>no information on numerical uptake values</li>
<li>no legend</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of
theoretical/experimental maximal uptake</li>
</ul>
</div>
<div class="section level2">
<h2 id="coverage-heatmap">Coverage heatmap<a class="anchor" aria-label="anchor" href="#coverage-heatmap"></a>
</h2>
<p>Coverage heatmap plot is a variation of standard coverage plot - but
with each peptide is colored to signal specific value. This plot is
particularly useful when presenting AUC or back-exchange values, as they
are specified for peptide uptake curve.</p>
<p><strong>Example</strong> Plot below presents the AUC values for
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>.
We see that for the majority of regions the AUC values is close to 1,
signifying fast exchange with exception for one strong region and small
sub-regions.</p>
<p>AUC:</p>
<pre><code><span><span class="co">## Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">HaDeX2</span> package.</span></span>
<span><span class="co">##   Please report the issue to the authors.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<pre><code><span><span class="co">## Ignoring unknown labels:</span></span>
<span><span class="co">## <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">colour</span> : <span style="color: #0000BB;">"Exposure"</span></span></span></code></pre>
<p><img src="visualization_files/figure-html/unnamed-chunk-15-1.png" class="r-plt" width="672"></p>
<p>Each of the AUC value for different biological state is calculated
separately. In order to meaningfully compare AUC values, we must select
the same fully deuterated control for all of them - to have the same
reference point of 100 % exchange. Otherwise, when the uptake curves
have different values in the plus infinity, the AUC values described
only the speed of the exchange taken separately. In the end, this value
is more useful when used to compare uptake curves for a specific peptide
under different biological conditions.</p>
<p><strong>Example</strong> The coverage heatmap plot below presents the
back-exchange values for peptides form
eEF1B<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>.
Back-exchange is believed to be on average close to 30%, as we see on
the plot. Some peptides - especially shorter ones - have grater
back-exchange.</p>
<pre><code><span><span class="co">## Ignoring unknown labels:</span></span>
<span><span class="co">## <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">colour</span> : <span style="color: #0000BB;">"Exposure"</span></span></span></code></pre>
<p><img src="visualization_files/figure-html/unnamed-chunk-16-1.png" class="r-plt" width="672"></p>
<p><em>Pros:</em></p>
<ul>
<li>auc / back-exchange values</li>
<li>one value for whole time course</li>
<li>peptide length and position</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>values may be hard to interpret within small range</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2">
<h2 id="summary-of-the-uptake-plots">Summary of the uptake plots<a class="anchor" aria-label="anchor" href="#summary-of-the-uptake-plots"></a>
</h2>
<p>Below we compare the aspects of the plots.</p>
<table class="table">
<colgroup>
<col width="21%">
<col width="11%">
<col width="20%">
<col width="11%">
<col width="12%">
<col width="15%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left">types</th>
<th align="left">time.course</th>
<th align="left">length.of.the.peptide</th>
<th align="left">uncertainty</th>
<th align="left">all.peptides</th>
<th align="left">different.states</th>
<th align="left">position</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">comparison</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Woods (differential)</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">butterfly</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">butterfly differential</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">volcano</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">chiclet</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">chiclet differential</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">uptake curve</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
<p>The columns indicate:</p>
<ul>
<li>time course - does this plot show the results from different time
points?</li>
<li>length of the peptide - does the plot show the information of the
length of the peptide?</li>
<li>uncertainty - does this plot show the uncertainty of the
measurement?</li>
<li>all peptides - does this plot show the results for all of the
peptides?</li>
<li>different states - does this plot show the results for different
states?</li>
</ul>
</div>
<div class="section level2">
<h2 id="new-features">New features<a class="anchor" aria-label="anchor" href="#new-features"></a>
</h2>
<p>We believe that new ways of visualization of the HDX-MS data can be
developed. If so, we will be happy to include them in HaDeX package and
update this article.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Weronika Puchala, Michal Burdukiewicz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
