<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Data visualization • HaDeX2</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Data visualization">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HaDeX2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/data_input.html">Importing data</a></li>
    <li><a class="dropdown-item" href="../articles/datafiles.html">Calculations</a></li>
    <li><a class="dropdown-item" href="../articles/example.html">Example HDX-MS data analysis</a></li>
    <li><a class="dropdown-item" href="../articles/glossary.html">Glossary</a></li>
    <li><a class="dropdown-item" href="../articles/mass_comparison.html">Mass calculation</a></li>
    <li><a class="dropdown-item" href="../articles/quality_control.html">Quality control</a></li>
    <li><a class="dropdown-item" href="../articles/statistics.html">Statistics</a></li>
    <li><a class="dropdown-item" href="../articles/transformation.html">Data aggregation</a></li>
    <li><a class="dropdown-item" href="../articles/usage.html">Code usage</a></li>
    <li><a class="dropdown-item" href="../articles/version_comparison.html">Comparison between versions</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Data visualization</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Data visualization</h1>
            
      

      <div class="d-none name"><code>visualization.Rmd</code></div>
    </div>

    
    
<p>Data visualization is a crucial part of the experimental data analysis. The forms of visualization should be adjusted to highlight the essential result and tailored to satisfy personal needs.</p>
<p>This article describes the visualization methods available in HaDeX2 and explains how different plot types can be used to interpret HDX-MS data. It focuses on the biological questions each visualization addresses rather than implementation details.</p>
<p>The analyzed protein is the eEF1B<span class="math inline">\(\alpha\)</span> subunit of the human guanine-nucleotide exchange factor (GEF) complex (eEF1B), measured in <a href="https://mslab-ibb.pl/" class="external-link">Mass Spectrometry Lab</a> in <a href="https://ibb.edu.pl/en/" class="external-link">Institute of Biochemistry and Biophysics Polish Academy of Sciences</a> <span class="citation">(Bondarchuk et al. 2022)</span>. In the one-state classification, we will focus on pure alpha state - eEF1B<span class="math inline">\(\alpha\)</span>. The comparative analysis is conducted between eEF1B<span class="math inline">\(\alpha\)</span> and eEF1B<span class="math inline">\(\alpha\)</span> in presence of eEF1B<span class="math inline">\(\gamma\)</span>.</p>
<div class="section level2" number="1">
<h2 id="comparison-plot">
<span class="header-section-number">1</span> Comparison plot<a class="anchor" aria-label="anchor" href="#comparison-plot"></a>
</h2>
<p>The comparison plot presents deuterium uptake of the peptides in a given time point, with information on the length of the peptide and their position in the protein sequence. It allows comparison of the results of different biological states.</p>
<p><strong>Example</strong> In the comparison plot below, we see the fractional deuterium uptake for all three possible states: eEF1B<span class="math inline">\(\alpha\)</span>, eEF1B<span class="math inline">\(\alpha\)</span> in presence of eEF1B<span class="math inline">\(\gamma\)</span> and in presence of eEF1B<span class="math inline">\(\beta\)</span>, from protein eEF1B. The values are calculated for the time point 1 min. The length of the segments represents the length of the peptide and the position in the protein sequence. The error bars indicate the uncertainty of the measurement.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/create_state_comparison_dataset.html">create_state_comparison_dataset</a></span><span class="op">(</span><span class="va">alpha_dat</span>, time_t <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="../reference/plot_state_comparison.html">plot_state_comparison</a></span><span class="op">(</span><span class="va">.</span>, fractional <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Position in sequence"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Fractional deuterium uptake [%]"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"Measurement after 1 min of exchange"</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-2-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>peptide length and position</li>
<li>comparison of multiple biological states</li>
<li>region with difference is easy to spot on</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
<li>plots with different time points can be plotted next to each other</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="2">
<h2 id="woods-plot">
<span class="header-section-number">2</span> Woods plot<a class="anchor" aria-label="anchor" href="#woods-plot"></a>
</h2>
<p>Woods plot presents the deuterium uptake difference between two biological states for the peptides. The results are presented with respect to the length of the peptide and its position in the protein sequence for a given time point of the measurement. The statistical test is applied to determine the confidence limits values at the chosen level. For more description see the <code><a href="../articles/statistics.html">vignette("statistics")</a></code>.</p>
<p><strong>Example</strong> On the Woods plot below, we see fractional deuterium uptake difference between two biological states eEF1B<span class="math inline">\(\alpha\)</span> and eEF1B<span class="math inline">\(\alpha\)</span> in presence of eEF1B<span class="math inline">\(\gamma\)</span> for protein eEF1B The confidence limits indicate which differences are statistically significant at levels 98%.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/calculate_diff_uptake.html">calculate_diff_uptake</a></span><span class="op">(</span><span class="va">alpha_dat</span>, states <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">states</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, <span class="va">states</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="../reference/plot_differential.html">plot_differential</a></span><span class="op">(</span><span class="va">.</span>, fractional <span class="op">=</span> <span class="cn">TRUE</span>, show_houde_interval <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Position in seqence"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Fractional deuterium uptake difference [%]"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"Measurement after 1 min of uptake"</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-3-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>differential analysis between two states</li>
<li>peptide length and position</li>
<li>region with difference is easy to spot on</li>
<li>uncertainty for each peptide</li>
<li>hybrid statistical testing</li>
<li>statistical level</li>
</ul>
<p><em>Plot variables:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
<li>plots with different time points can be plotted next to each other</li>
<li>peptides classified as statistically insignificant can be hidden</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="3">
<h2 id="butterfly-plot">
<span class="header-section-number">3</span> Butterfly plot<a class="anchor" aria-label="anchor" href="#butterfly-plot"></a>
</h2>
<p>Butterfly plot presents the deuterium uptake for all peptides in a given state at different time points at once. Each time point of measurement is indicated by a different color. Peptides are identified by their ID (peptides are numbered arranged by the start position).</p>
<p><strong>Example</strong> Below, on the butterfly plot, we see how the deuterium uptake changes in time for state eEF1B<span class="math inline">\(\alpha\)</span> for protein eEF1B. We see the different exchange speed - for some peptides, the change is stable in time, and for some peptides, there is no visible change in time.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/create_state_uptake_dataset.html">create_state_uptake_dataset</a></span><span class="op">(</span><span class="va">alpha_dat</span>, state <span class="op">=</span> <span class="va">states</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="../reference/plot_butterfly.html">plot_butterfly</a></span><span class="op">(</span><span class="va">.</span>, fractional <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-4-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>values for time course</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>only one state</li>
<li>lost information of length and position of the peptide</li>
<li>may be difficult to read</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
<li>different methods of showing uncertainty: bars or ribbons</li>
<li>selected time points shown</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="4">
<h2 id="butterfly-differential-plot">
<span class="header-section-number">4</span> Butterfly differential plot<a class="anchor" aria-label="anchor" href="#butterfly-differential-plot"></a>
</h2>
<p>Butterfly differential plot shows the deuterium uptake difference between two biological states in the form of a butterfly plot. It shows the results for a peptide ID (peptides are numbered arranged by the start position). The results are shown for different time points at once (time points of measurement are indicated by the color).</p>
<p><strong>Example</strong> Below, we see how the fractional deuterium uptake difference between states eEF1B<span class="math inline">\(\alpha\)</span> and eEF1B<span class="math inline">\(\alpha\)</span> in presence of eEF1B<span class="math inline">\(\gamma\)</span> changes over time. We see that for some peptides, the difference is smaller with time - perhaps because of the back exchange.</p>
<p>The measurements for 1440 min are hidden, as they are close to 0, as expected.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/create_diff_uptake_dataset.html">create_diff_uptake_dataset</a></span><span class="op">(</span><span class="va">alpha_dat</span>, state_1 <span class="op">=</span> <span class="va">states</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, state_2 <span class="op">=</span> <span class="va">states</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Exposure</span> <span class="op">&lt;</span> <span class="fl">1440</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="../reference/plot_differential_butterfly.html">plot_differential_butterfly</a></span><span class="op">(</span>fractional <span class="op">=</span> <span class="cn">TRUE</span>, show_houde_interval <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-5-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>differential analysis between two states</li>
<li>hybrid statistical testing</li>
<li>statistical significance level</li>
<li>values for time course</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>lost information of length and position of the peptide</li>
<li>may be difficult to read</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
<li>different methods of showing uncertainty: bars or ribbons</li>
<li>selected time points shown</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="5">
<h2 id="chiclet-plot">
<span class="header-section-number">5</span> Chiclet plot<a class="anchor" aria-label="anchor" href="#chiclet-plot"></a>
</h2>
<p>Chiclet plot shows the fractional deuterium uptake in the form of a heatmap for the peptides in a given biological state. One tile indicates the peptide (identified by its ID - number arranged by the start position) in a time point of measurement. The color of the tile indicates the fractional deuterium uptake (according to the legend below the plot).</p>
<p><strong>Example</strong> In the chiclet plot below, we can see the deuterium uptake values for peptides (indicated by their ID) in state eEF1B<span class="math inline">\(\alpha\)</span> during the time course of the experiment. The cross symbols indicate the uncertainty of the measurement (the bigger the cross sign, the bigger the uncertainty).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/create_state_uptake_dataset.html">create_state_uptake_dataset</a></span><span class="op">(</span><span class="va">alpha_dat</span>, state <span class="op">=</span> <span class="va">states</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Exposure</span> <span class="op">&lt;</span> <span class="fl">1440</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="../reference/plot_chiclet.html">plot_chiclet</a></span><span class="op">(</span>show_uncertainty <span class="op">=</span> <span class="cn">TRUE</span>, fractional <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-6-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>values for time course</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>only one state</li>
<li>lost information of length and position of the peptide</li>
<li>may be difficult to read</li>
<li>small changes in values may be difficult to spot</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
<li>uncertainty can be hidden to improve the readability of the plot</li>
<li>peptides classified as statistically significant can be hidden</li>
<li>selected time points shown</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="6">
<h2 id="chiclet-differential-plot">
<span class="header-section-number">6</span> Chiclet differential plot<a class="anchor" aria-label="anchor" href="#chiclet-differential-plot"></a>
</h2>
<p>Chiclet differential plot shows the deuterium uptake difference between two biological states in the form of a heatmap. One tile indicates the peptide (identified by its ID - number arranged by the start position) in a time point of measurement. The color of the tile indicates the deuterium uptake difference (according to the legend below the plot).</p>
<p><strong>Example</strong> On the chiclet differential plot below, we see the fractional deuterium uptake difference between states eEF1B<span class="math inline">\(\alpha\)</span> and eEF1B<span class="math inline">\(\alpha\)</span> in presence of eEF1B<span class="math inline">\(\gamma\)</span> for protein eEF1B. We see that some peptides are protected (red), and some are deprotected (blue). The cross symbols indicate the uncertainty of the measurement (the bigger the cross sign, the bigger the uncertainty).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_uptake_dat</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Exposure</span> <span class="op">&lt;</span> <span class="fl">1440</span> <span class="op">&amp;</span> <span class="va">Exposure</span> <span class="op">&gt;</span> <span class="fl">0.001</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="../reference/plot_differential_chiclet.html">plot_differential_chiclet</a></span><span class="op">(</span>show_uncertainty <span class="op">=</span> <span class="cn">TRUE</span>, fractional <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-7-1.png"><!-- --><em>Pros:</em></p>
<ul>
<li>differential analysis between two states</li>
<li>values for time course</li>
<li>uncertainty for each peptide</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>lost information of length and position of the peptide</li>
<li>may be difficult to read</li>
<li>small changes in values may be difficult to spot</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
<li>uncertainty can be hidden to improve the readability of the plot</li>
<li>peptides classified as statistically significant can be hidden</li>
<li>selected time points shown</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="7">
<h2 id="volcano-plot">
<span class="header-section-number">7</span> Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h2>
<p>The volcano plot shows the deuterium uptake difference for two biological states for peptide and its p-value for double testing on statistical significance <span class="citation">(Hageman and Weis 2019)</span>. On the x-axis, there is a deuterium uptake difference with its uncertainty (combined and propagated). On the y-axis, there is a P-value calculated for each peptide in a specific time point of a measurement as a un-paired t-test on given significance level (on mass measurement from the replicates to indicate if the measured mean is significantly different between two states, as the deuterium uptake difference between states can be rewritten as</p>
<p><span class="math display">\[\Delta D = D_{A} - D_{B} = m_{t, A} - m_{0} - (m_{t, B} - m_{0}) = m_{t, A} - m_{t, B} \]</span></p>
<p>for states A and B. The values of deuterium uptake difference from all time points are shown on the plot.</p>
<p>The dotted red lines indicate confidence limits for the values. The horizontal line indicates the confidence limit based on chosen confidence level to give a threshold on a P-value. The vertical lines indicate the confidence limit from Houde test for all time points and indicate a threshold on deuterium uptake difference. The statistically significant points are in the top left and right corners of the plot.</p>
<p><strong>Example</strong> On the volcano plot below, we see the results for deuterium uptake difference between states eEF1B<span class="math inline">\(\alpha\)</span> and eEF1B<span class="math inline">\(\alpha\)</span> in presence of eEF1B<span class="math inline">\(\gamma\)</span> for protein eEF1B in all time points. The points in the left and right upper corner are statistically significant using the hybrid testing.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_p_diff_uptake_dataset.html">create_p_diff_uptake_dataset</a></span><span class="op">(</span><span class="va">alpha_dat</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">p_dat</span>, show_confidence_limits <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-8-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>hybrid statistic test</li>
<li>time course information</li>
<li>uncertainty</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>peptide length and position lost</li>
<li>no information of region</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>selected time points</li>
<li>absolute/fractional values s</li>
<li>hidden insignificant values</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="8">
<h2 id="uptake-curve">
<span class="header-section-number">8</span> Uptake curve<a class="anchor" aria-label="anchor" href="#uptake-curve"></a>
</h2>
<p>Uptake curves show the changes in exchange in time for a specific peptide for its state.</p>
<p><strong>Example</strong> On the uptake curve below, we see how the exchange goes for peptide GFGDLKSPAGL in all three states for protein eEF1Ba.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/calculate_peptide_kinetics.html">calculate_peptide_kinetics</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">alpha_dat</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span><span class="fu"><a href="../reference/plot_uptake_curve.html">plot_uptake_curve</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ylim</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-9-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>time course</li>
<li>state comparison</li>
<li>change of uptake tendency visible</li>
<li>focus on one peptide</li>
<li>uncertainty</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
<li>selected states shown</li>
<li>different methods of showing uncertainty: bars or ribbons</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="9">
<h2 id="uncertainty-plot">
<span class="header-section-number">9</span> Uncertainty plot<a class="anchor" aria-label="anchor" href="#uncertainty-plot"></a>
</h2>
<p>Uncertainty plot is new visualization method, showing the uncertainty of measurement of deuterium uptake for peptides to spot the regions where the uncertainty is higher. This plot may be used as a quality control of the experiment, as discussed in the <code><a href="../articles/quality_control.html">vignette("quality_control")</a></code>.
The presented uncertainty is in Daltons, making the threshold 1 Da as proposed limit of acceptance.</p>
<p><strong>Example</strong> The plot below presents uncertainty values for multiple time points for state eEF1B<span class="math inline">\(\alpha\)</span>. We see that the uncertainty is relatively low for all the measurement, and none of the value is suspicious.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alpha_dat</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Exposure</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="../reference/plot_uncertainty.html">plot_uncertainty</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-10-1.png"><!-- --><em>Pros:</em></p>
<ul>
<li>experimental quality control</li>
<li>regions with high measurement uncertainty are easy to spot</li>
<li>peptide length and position</li>
<li>region with difference is easy to spot on</li>
</ul>
<p><em>Plot variants:</em></p>
<ul>
<li>plot for only one time point can be plotted</li>
<li>selected time points shown</li>
<li>threshold can be hidden</li>
<li>first amino omitted</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="10">
<h2 id="manhattan-plot">
<span class="header-section-number">10</span> Manhattan plot<a class="anchor" aria-label="anchor" href="#manhattan-plot"></a>
</h2>
<p>Manhattan plot is a novel plot, presenting the P-value of statistical significance between two states.</p>
<p><strong>Example</strong> In this example, we present the P values of difference between two biological states: eEF1B<span class="math inline">\(\alpha\)</span> and eEF1B<span class="math inline">\(\alpha\)</span> in presence of eEF1B<span class="math inline">\(\gamma\)</span>.
We can see the regions where the difference is statistically significant - above the significance level (detailed as option in create_p_diff_uptake_dataset, or by default 0.98).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p_diff_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_p_diff_uptake_dataset.html">create_p_diff_uptake_dataset</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">alpha_dat</span>, diff_uptake_dat <span class="op">=</span> <span class="va">diff_uptake_dat</span>,</span>
<span>                                           state_1 <span class="op">=</span> <span class="va">states</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, state_2 <span class="op">=</span> <span class="va">states</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_manhattan.html">plot_manhattan</a></span><span class="op">(</span><span class="va">p_diff_dat</span>, show_peptide_position  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-11-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>peptide length and position</li>
<li>shows significant regions</li>
<li>statistical interval shown</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>plot for only one time point can be plotted</li>
<li>selected time points shown</li>
<li>threshold can be hidden</li>
<li>first amino omitted</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="11">
<h2 id="high-resolution-plot">
<span class="header-section-number">11</span> High-resolution plot<a class="anchor" aria-label="anchor" href="#high-resolution-plot"></a>
</h2>
<p>The biggest limitation of previous methods of deuterium uptake visualization is that the results are on the peptide level. However, we offer a method of deuterium uptake averaging from peptide level into high-resolution level, using the weighted method of averaging (as described in <code><a href="../articles/statistics.html">vignette("statistics")</a></code>). Then, the results are presented on the heatmap.</p>
<p><strong>Example</strong></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kin_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_uptake_dataset.html">create_uptake_dataset</a></span><span class="op">(</span><span class="va">alpha_dat</span>, states <span class="op">=</span> <span class="st">"Alpha_KSCN"</span><span class="op">)</span></span>
<span><span class="va">aggregated_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_aggregated_uptake_dataset.html">create_aggregated_uptake_dataset</a></span><span class="op">(</span><span class="va">kin_dat</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_aggregated_uptake.html">plot_aggregated_uptake</a></span><span class="op">(</span><span class="va">aggregated_dat</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-12-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>high-resolution</li>
<li>time course</li>
<li>uncertainty of averaged values</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>values are an approximation</li>
<li>linear presentation lacking conformational information</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
<li>plot divided into panels</li>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="12">
<h2 id="differential-high-resolution-plot">
<span class="header-section-number">12</span> Differential High-resolution plot<a class="anchor" aria-label="anchor" href="#differential-high-resolution-plot"></a>
</h2>
<p>As most of our methods of visualization, also the high-resolution plot has its differential version. The plot presents averaged uptake difference values using weighted approach.</p>
<p><strong>Example</strong> On the structure, we see the fractional deuterium uptake difference after 25 minutes between states eEF1B<span class="math inline">\(\alpha\)</span> and eEF1B<span class="math inline">\(\alpha\)</span> in presence of eEF1B<span class="math inline">\(\gamma\)</span> for protein eEF1B. We see that some peptides are protected (red), and some are deprotected (blue).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_uptake_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_diff_uptake_dataset.html">create_diff_uptake_dataset</a></span><span class="op">(</span><span class="va">alpha_dat</span>, state_1 <span class="op">=</span> <span class="va">states</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, state_2 <span class="op">=</span> <span class="va">states</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">averaged_diff_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_aggregated_diff_uptake_dataset.html">create_aggregated_diff_uptake_dataset</a></span><span class="op">(</span><span class="va">diff_uptake_dat</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_aggregated_differential_uptake.html">plot_aggregated_differential_uptake</a></span><span class="op">(</span><span class="va">averaged_diff_dat</span>, panels <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-13-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>high-resolution</li>
<li>comparative analysis</li>
<li>time course</li>
<li>uncertainty of averaged values</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>values are an approximation</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
<li>plot divided into panels</li>
<li>tooltips available in GUI</li>
</ul>
<!-- # Differential High-resolution on 3D structure --><!-- Differential aggregated results can be presented on 3d structure, if available. This plot has a very limited options for personalization, but is extremly vital for comparison of experimental uptake data with spatial information about conformation of the protein. --><!-- **Example**  On the structure, we see the fractional deuterium uptake difference after 25 minutes between states eEF1B$\alpha$ and eEF1B$\alpha$ in presence of eEF1B$\gamma$ for protein eEF1B. We see that some peptides are protected (red), and some are deprotected (blue). --><!-- ```{r} --><!-- pdb_file_path <- system.file(package = "HaDeX", "HaDeX/data/Model_eEF1Balpha.pdb") --><!-- plot_aggregated_uptake_structure(aggregated_diff_dat,  --><!--                                  differential = TRUE, --><!--                                  time_t = 25, --><!--                                  pdb_file_path = pdb_file_path) --><!-- ``` --><!-- *Pros:* --><!--  * aggregated values --><!--  * spatial information --><!--  * spinning, rotating and zooming options --><!-- *Cons:* --><!--  * only one time point at a time --><!--  * no personalization --><!--  * no information on numerical uptake values --><!--  * no legend --><!-- *Plot variations:* --><!--  * fractional/absolute values --><!--  * fractional values calculated with respect of theoretical/experimental maximal uptake -->
</div>
<div class="section level2" number="13">
<h2 id="high-resolution-on-3d-structure">
<span class="header-section-number">13</span> High-resolution on 3D structure<a class="anchor" aria-label="anchor" href="#high-resolution-on-3d-structure"></a>
</h2>
<p>High-resolution values not only can be presented linearly on high-resolution plot (as above), but also on 3D structure, if available. This way, the calculated values are connected with spatial aspect. This option is available both for single state uptake and differential uptake.</p>
<p><strong>Example</strong> Structure below is mapped with deuterium uptake values after 1 minute for eEF1B<span class="math inline">\(\alpha\)</span> state. Values are averaged using weighted approach, color signifies no exchange (white) to high exchange (red).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pdb_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"HaDeX2"</span>, <span class="st">"HaDeX/data/Model_eEF1Balpha.pdb"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_aggregated_uptake_structure.html">plot_aggregated_uptake_structure</a></span><span class="op">(</span><span class="va">aggregated_dat</span>, </span>
<span>                                 differential <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 time_t <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                 pdb_file_path <span class="op">=</span> <span class="va">pdb_file_path</span><span class="op">)</span></span></code></pre></div>
<p><em>Pros:</em></p>
<ul>
<li>averaged values</li>
<li>spatial information</li>
<li>spinning, rotating and zooming options</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>only one time point at a time</li>
<li>no personalization</li>
<li>no information on numerical uptake values</li>
<li>no legend</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>fractional/absolute values</li>
<li>fractional values calculated with respect of theoretical/experimental maximal uptake</li>
</ul>
</div>
<div class="section level2" number="14">
<h2 id="coverage-heatmap">
<span class="header-section-number">14</span> Coverage heatmap<a class="anchor" aria-label="anchor" href="#coverage-heatmap"></a>
</h2>
<p>Coverage heatmap plot is a variation of standard coverage plot - but with each peptide is colored to signal specific value. This plot is particularly useful when presenting AUC (consult <a href="#AUC"><strong>??</strong></a> for the details of AUC computation) or back-exchange values, as they are specified for peptide uptake curve.</p>
<p><strong>Example</strong> Plot below presents the AUC values for eEF1B<span class="math inline">\(\alpha\)</span>. We see that for the majority of regions the AUC values is close to 1, signifying fast exchange with exception for one strong region and small sub-regions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">auc_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_auc.html">calculate_auc</a></span><span class="op">(</span><span class="fu"><a href="../reference/create_uptake_dataset.html">create_uptake_dataset</a></span><span class="op">(</span><span class="va">alpha_dat</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_coverage_heatmap.html">plot_coverage_heatmap</a></span><span class="op">(</span><span class="va">auc_dat</span>, value <span class="op">=</span> <span class="st">"auc"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Ignoring unknown labels:</span></span>
<span><span class="co">## <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">colour</span> : <span style="color: #0000BB;">"Exposure"</span></span></span></code></pre>
<p><img src="visualization_files/figure-html/unnamed-chunk-15-1.png"><!-- --></p>
<p>Each of the AUC value for different biological state is calculated separately. In order to meaningfully compare AUC values, we must select the same fully deuterated control for all of them - to have the same reference point of 100 % exchange. Otherwise, when the uptake curves have different values in the plus infinity, the AUC values described only the speed of the exchange taken separately. In the end, this value is more useful when used to compare uptake curves for a specific peptide under different biological conditions.</p>
<p><strong>Example</strong> The coverage heatmap plot below presents the back-exchange values for peptides form eEF1B<span class="math inline">\(\alpha\)</span>. Back-exchange is believed to be on average close to 30%, as we see on the plot. Some peptides - especially shorter ones - have grater back-exchange.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bex_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_back_exchange.html">calculate_back_exchange</a></span><span class="op">(</span><span class="va">alpha_dat</span>, state <span class="op">=</span> <span class="st">"Alpha_KSCN"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_coverage_heatmap.html">plot_coverage_heatmap</a></span><span class="op">(</span><span class="va">bex_dat</span>, value <span class="op">=</span> <span class="st">"back_exchange"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Ignoring unknown labels:</span></span>
<span><span class="co">## <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">colour</span> : <span style="color: #0000BB;">"Exposure"</span></span></span></code></pre>
<p><img src="visualization_files/figure-html/unnamed-chunk-16-1.png"><!-- --></p>
<p><em>Pros:</em></p>
<ul>
<li>auc / back-exchange values</li>
<li>one value for whole time course</li>
<li>peptide length and position</li>
</ul>
<p><em>Cons:</em></p>
<ul>
<li>values may be hard to interpret within small range</li>
</ul>
<p><em>Plot variations:</em></p>
<ul>
<li>tooltips available in GUI</li>
</ul>
</div>
<div class="section level2" number="15">
<h2 id="summary-of-the-uptake-plots">
<span class="header-section-number">15</span> Summary of the uptake plots<a class="anchor" aria-label="anchor" href="#summary-of-the-uptake-plots"></a>
</h2>
<p>Below we compare the aspects of the plots.</p>
<table class="table">
<colgroup>
<col width="21%">
<col width="11%">
<col width="20%">
<col width="11%">
<col width="12%">
<col width="15%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left">types</th>
<th align="left">time course</th>
<th align="left">length of the peptide</th>
<th align="left">uncertainty</th>
<th align="left">all peptides</th>
<th align="left">different states</th>
<th align="left">position</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">comparison</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Woods (differential)</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">butterfly</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">butterfly differential</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">volcano</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">chiclet</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">chiclet differential</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">uptake curve</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
<p>The columns indicate:</p>
<ul>
<li>time course - does this plot show the results from different time points?</li>
<li>length of the peptide - does the plot show the information of the length of the peptide?</li>
<li>uncertainty - does this plot show the uncertainty of the measurement?</li>
<li>all peptides - does this plot show the results for all of the peptides?</li>
<li>different states - does this plot show the results for different states?</li>
</ul>
<!-- # New features --><!-- We believe that new ways of visualization of the HDX-MS data can be developed. If so, we will be happy to include them in HaDeX package and update this article.  --><div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-bondarchuk_quaternary_2022" class="csl-entry">
Bondarchuk, Tetiana V, Vyacheslav F Shalak, Dmytro M Lozhko, Agnieszka Fatalska, Roman H Szczepanowski, Vladyslava Liudkovska, Oleksandr Yu Tsuvariev, Michal Dadlez, Anna V El’skaya, and Boris S Negrutskii. 2022. <span>“Quaternary Organization of the Human <span class="nocase">eEF1B</span> Complex Reveals Unique Multi-<span>GEF</span> Domain Assembly.”</span> <em>Nucleic Acids Research</em> 50 (16): 9490–9504. <a href="https://doi.org/10.1093/nar/gkac685" class="external-link">https://doi.org/10.1093/nar/gkac685</a>.
</div>
<div id="ref-hageman_reliable_2019" class="csl-entry">
Hageman, Tyler S., and David D. Weis. 2019. <span>“Reliable <span>Identification</span> of <span>Significant</span> <span>Differences</span> in <span>Differential</span> <span>Hydrogen</span> <span>Exchange</span>-<span>Mass</span> <span>Spectrometry</span> <span>Measurements</span> <span>Using</span> a <span>Hybrid</span> <span>Significance</span> <span>Testing</span> <span>Approach</span>.”</span> <em>Analytical Chemistry</em> 91 (13): 8008–16. <a href="https://doi.org/10.1021/acs.analchem.9b01325" class="external-link">https://doi.org/10.1021/acs.analchem.9b01325</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Weronika Puchala, Michal Burdukiewicz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
